<!DOCTYPE html>
<html lang="en-US">
  <head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width,initial-scale=1">
    <title>SV Calling | Variant calling for NGS data</title>
    <meta name="description" content="A tutorial on variant calling for NGS data using FreeBayes and Delly">
    
    
    <link rel="preload" href="/courses/vc/assets/css/0.styles.a71e43cb.css" as="style"><link rel="preload" href="/courses/vc/assets/js/app.eb2d6309.js" as="script"><link rel="preload" href="/courses/vc/assets/js/2.65bda2c6.js" as="script"><link rel="preload" href="/courses/vc/assets/js/9.119cb8a5.js" as="script"><link rel="prefetch" href="/courses/vc/assets/js/10.1a851d49.js"><link rel="prefetch" href="/courses/vc/assets/js/3.216ec158.js"><link rel="prefetch" href="/courses/vc/assets/js/4.732b70ac.js"><link rel="prefetch" href="/courses/vc/assets/js/5.8fd54cbc.js"><link rel="prefetch" href="/courses/vc/assets/js/6.33be47b5.js"><link rel="prefetch" href="/courses/vc/assets/js/7.2e70c853.js"><link rel="prefetch" href="/courses/vc/assets/js/8.ff6ee130.js">
    <link rel="stylesheet" href="/courses/vc/assets/css/0.styles.a71e43cb.css">
  </head>
  <body>
    <div id="app" data-server-rendered="true"><div class="theme-container"><header class="navbar"><div class="sidebar-button"><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" role="img" viewBox="0 0 448 512" class="icon"><path fill="currentColor" d="M436 124H12c-6.627 0-12-5.373-12-12V80c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12zm0 160H12c-6.627 0-12-5.373-12-12v-32c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12zm0 160H12c-6.627 0-12-5.373-12-12v-32c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12z"></path></svg></div> <a href="/courses/vc/" class="home-link router-link-active"><!----> <span class="site-name">Variant calling for NGS data</span></a> <div class="links"><div class="search-box"><input aria-label="Search" autocomplete="off" spellcheck="false" value=""> <!----></div> <nav class="nav-links can-hide"><div class="nav-item"><a href="/courses/vc/" class="nav-link">Home</a></div><div class="nav-item"><a href="/courses/vc/snv/" class="nav-link">SNV Calling</a></div><div class="nav-item"><a href="/courses/vc/sv/" class="nav-link router-link-exact-active router-link-active">SV Calling</a></div><div class="nav-item"><a href="/courses/vc/appendix/" class="nav-link">Appendix</a></div> <a href="https://github.com/tobiasrausch/variant-calling" target="_blank" rel="noopener noreferrer" class="repo-link">
    GitHub
    <svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg></a></nav></div></header> <div class="sidebar-mask"></div> <aside class="sidebar"><nav class="nav-links"><div class="nav-item"><a href="/courses/vc/" class="nav-link">Home</a></div><div class="nav-item"><a href="/courses/vc/snv/" class="nav-link">SNV Calling</a></div><div class="nav-item"><a href="/courses/vc/sv/" class="nav-link router-link-exact-active router-link-active">SV Calling</a></div><div class="nav-item"><a href="/courses/vc/appendix/" class="nav-link">Appendix</a></div> <a href="https://github.com/tobiasrausch/variant-calling" target="_blank" rel="noopener noreferrer" class="repo-link">
    GitHub
    <svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg></a></nav>  <ul class="sidebar-links"><li><a href="/courses/vc/snv/" class="sidebar-link">SNV Calling</a></li><li><a href="/courses/vc/sv/" class="active sidebar-link">SV Calling</a><ul class="sidebar-sub-headers"><li class="sidebar-sub-header"><a href="/courses/vc/sv/#introduction" class="sidebar-link">Introduction</a></li><li class="sidebar-sub-header"><a href="/courses/vc/sv/#germline-deletions" class="sidebar-link">Germline Deletions</a></li><li class="sidebar-sub-header"><a href="/courses/vc/sv/#interactive-igv" class="sidebar-link">Interactive IGV</a></li><li class="sidebar-sub-header"><a href="/courses/vc/sv/#igv-in-batch-mode" class="sidebar-link">IGV in batch mode</a></li><li class="sidebar-sub-header"><a href="/courses/vc/sv/#complex-germline-structural-variants" class="sidebar-link">Complex Germline Structural Variants</a></li><li class="sidebar-sub-header"><a href="/courses/vc/sv/#structural-variant-alignment-qc" class="sidebar-link">Structural Variant Alignment QC</a></li><li class="sidebar-sub-header"><a href="/courses/vc/sv/#structural-variant-calling" class="sidebar-link">Structural Variant Calling</a></li><li class="sidebar-sub-header"><a href="/courses/vc/sv/#somatic-filtering" class="sidebar-link">Somatic Filtering</a></li><li class="sidebar-sub-header"><a href="/courses/vc/sv/#structural-variant-visualization-using-igv" class="sidebar-link">Structural Variant Visualization using IGV</a></li><li class="sidebar-sub-header"><a href="/courses/vc/sv/#interactive-igv-2" class="sidebar-link">Interactive IGV</a></li><li class="sidebar-sub-header"><a href="/courses/vc/sv/#igv-in-batch-mode-2" class="sidebar-link">IGV in batch mode</a></li><li class="sidebar-sub-header"><a href="/courses/vc/sv/#visualizing-complex-rearrangements" class="sidebar-link">Visualizing Complex Rearrangements</a></li><li class="sidebar-sub-header"><a href="/courses/vc/sv/#allelic-depth-for-snps" class="sidebar-link">Allelic Depth for SNPs</a></li></ul></li><li><a href="/courses/vc/appendix/" class="sidebar-link">Appendix</a></li></ul> </aside> <main class="page"> <div class="theme-default-content content__default"><h1 id="sv-calling"><a href="#sv-calling" aria-hidden="true" class="header-anchor">#</a> SV Calling</h1> <h2 id="introduction"><a href="#introduction" aria-hidden="true" class="header-anchor">#</a> Introduction</h2> <h3 id="discovery-of-chromothripsis-in-a-cancer-sample"><a href="#discovery-of-chromothripsis-in-a-cancer-sample" aria-hidden="true" class="header-anchor">#</a> Discovery of chromothripsis in a cancer sample</h3> <p>In this practical we will analyze germline and somatic structural variants (SVs) of a Chromothripsis sample from this <a href="https://www.ncbi.nlm.nih.gov/pubmed/22265402" target="_blank" rel="noopener noreferrer">study<svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg></a>. The anonomyzed data has been downsampled to chr2 to speed up all analysis hereafter. The alignment file of the tumor genome is called tumor.bam and the alignment file of the control genome is called control.bam.</p> <h2 id="germline-deletions"><a href="#germline-deletions" aria-hidden="true" class="header-anchor">#</a> Germline Deletions</h2> <p>To get an idea of how structural variants (SVs) look like in short read, next-generation sequencing data one can use the <a href="https://www.ncbi.nlm.nih.gov/pubmed/26432246" target="_blank" rel="noopener noreferrer">1000 Genomes SV polymorphism catalogue<svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg></a> because these are germline SVs and a subset of those will be present in our sample. To increase the chance that our sample of interest is a carrier of one of these germline SVs I have already subsetted the SV list to common SVs with a population allele frequency above 50% on chr2. The below <a href="https://genome.ucsc.edu/FAQ/FAQformat.html#format1" target="_blank" rel="noopener noreferrer">BED<svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg></a> file shows a couple of germline SVs.</p> <div class="language-bash extra-class"><pre class="language-bash"><code><span class="token function">cd</span> /data/sv/
<span class="token function">cat</span> svs.bed
</code></pre></div><p>Using <a href="http://software.broadinstitute.org/software/igv/" target="_blank" rel="noopener noreferrer">IGV<svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg></a> we can then browse these SVs interactively if an X11 display is available. If you only have a terminal we can still use IGV in batch mode and create snapshots of the putative SV loci.</p> <h2 id="interactive-igv"><a href="#interactive-igv" aria-hidden="true" class="header-anchor">#</a> Interactive IGV</h2> <p>With an X11 display we can start <a href="http://software.broadinstitute.org/software/igv/" target="_blank" rel="noopener noreferrer">IGV<svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg></a> and browse the variants interactively using the chr2 reference sequence.</p> <div class="language-bash extra-class"><pre class="language-bash"><code>./igv.sh -g chr2.fa
</code></pre></div><p>Once IGV has started use 'File' and 'Load from File' to load the control.bam alignment file. Then import 'svs.bed' from your working directory using 'Regions' and 'Import Regions'. The structural variants can then be browsed easily using 'Regions' and 'Region Navigator'. Select a structural variant in the Region Navigator and click 'View', which will center the IGV alignment view on the selected structural variant. It's usually best to zoom out once then by clicking on the '-' sign in the toolbar at the top, so you can view all supporting abnormal paired-ends. To highlight the abnormal paired-ends please right click in IGV on the bam file and activate 'View as pairs'. In the same menu, please open 'Color alignments by' and then switch to &quot;pair orientation' for inversions and duplications. For deletions, you want to color the alignments by &quot;insert size&quot;.</p> <h2 id="igv-in-batch-mode"><a href="#igv-in-batch-mode" aria-hidden="true" class="header-anchor">#</a> IGV in batch mode</h2> <p>Without an X11 display we can still use IGV to plot snapshots of the putative SVs.</p> <div class="language-bash extra-class"><pre class="language-bash"><code>./runIGVBatch.sh svs.bed
</code></pre></div><p>Once IGV is finished we can simply browse these snapshots. Some clear germline deletions are for instance.</p> <div class="language-bash extra-class"><pre class="language-bash"><code><span class="token function">open</span> DEL11.png
<span class="token function">open</span> DEL13.png
</code></pre></div><h2 id="complex-germline-structural-variants"><a href="#complex-germline-structural-variants" aria-hidden="true" class="header-anchor">#</a> Complex Germline Structural Variants</h2> <p>Most 1000 Genomes SVs are deletions because these are easier to detect in low coverage sequencing data. Deletions cause a drop in read-depth and can be detected by spanning paired-ends of abnormally large insert size. In addition, split-reads have a prefix alignment before the deletion and a suffix alignment after the deletion. Other SV types such as inversions are much harder to detect because these are balanced rearrangements that do not cause a read-depth change. Besides the simple SV types (deletions, duplications, inversions) one can also find more complex SVs in the germline. Two example regions for that are in the file complex.bed.</p> <div class="language-bash extra-class"><pre class="language-bash"><code>./runIGVBatch.sh complex.bed
<span class="token function">open</span> complex1.png
<span class="token function">open</span> complex2.png
</code></pre></div><p>As part of the SV consortium we validated the above complex SVs using PacBio. The reads are in a separate FASTA file called pacbio.sv1.fa and pacbio.sv2.fa. We need the subsequence of the reference to create a pairwise dotplot of the PacBio read against the reference. <a href="http://www.htslib.org" target="_blank" rel="noopener noreferrer">SAMtools<svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg></a> is a convenient tool to extract such subsequences of a FASTA file.</p> <div class="language-bash extra-class"><pre class="language-bash"><code><span class="token function">cat</span> complex.bed
samtools faidx chr2.fa chr2:18905691-18907969
samtools faidx chr2.fa chr2:96210505-96212783
</code></pre></div><p>Please align the above genomic reference subsequences against the respective PacBio read using <a href="https://gear.embl.de/maze/" target="_blank" rel="noopener noreferrer">Maze<svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg></a> available on <a href="https://gear.embl.de" target="_blank" rel="noopener noreferrer">gear.embl.de<svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg></a>.</p> <p><em><strong>Exercises</strong></em></p> <ul><li>What kind of SV is present in the region chr2:18905691-18907969 ?</li> <li>What kind of SV is present in the region chr2:96210505-96212783 ?</li></ul> <h2 id="structural-variant-alignment-qc"><a href="#structural-variant-alignment-qc" aria-hidden="true" class="header-anchor">#</a> Structural Variant Alignment QC</h2> <p>Paired-end methods can be affected by a skewed insert size distribution, read-depth methods by non-uniform coverage and split-read methods suffer from high sequencing error rates that cause mis-mappings. Prior to any structural variant discovery you should therefore evaluate the quality of the data such as the percentage of mapped reads, singletons, duplicates, properly paired reads and the insert size &amp; coverage distributions. <a href="http://broadinstitute.github.io/picard/" target="_blank" rel="noopener noreferrer">Picard<svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg></a>, <a href="http://www.htslib.org" target="_blank" rel="noopener noreferrer">SAMtools<svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg></a>, <a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/" target="_blank" rel="noopener noreferrer">FastQC<svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg></a> and <a href="https://github.com/tobiasrausch/alfred" target="_blank" rel="noopener noreferrer">Alfred<svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg></a> compute some of these alignment statistics as shown below for the tumor sample.</p> <div class="language-bash extra-class"><pre class="language-bash"><code>samtools flagstat tumor.bam
alfred qc -r chr2.fa tumor.bam
zcat qc.tsv.gz <span class="token operator">|</span> <span class="token function">grep</span> ^ME <span class="token operator">|</span> datamash transpose <span class="token operator">|</span> column -t
Rscript /opt/alfred/R/stats.R qc.tsv.gz
convert qc.tsv.gz.pdf qc.png
<span class="token comment"># Coverage distribution</span>
<span class="token function">open</span> qc-4.png
<span class="token comment"># Insert size distribution</span>
<span class="token function">open</span> qc-5.png
</code></pre></div><p>There are some general things to watch out for such as mapping percentages below 70%, &gt;20% duplicates or multiple peaks in the insert size distribution. However, these statistics vary largely by protocol and hence, it's usually best to compare multiple different sequencing runs using the same protocol (DNA-seq, RNA-seq, ChIP-seq, paired-end, single-end or mate-pair) against each other, which then highlights the outliers.</p> <p><em><strong>Exercises</strong></em></p> <ul><li>What is the median coverage of the data set?</li> <li>Given the insert size distribution, what would be a suitable cutoff to define deletion supporting paired-ends?</li></ul> <h2 id="structural-variant-calling"><a href="#structural-variant-calling" aria-hidden="true" class="header-anchor">#</a> Structural Variant Calling</h2> <p>As a first step we will compute structural variants using <a href="https://github.com/dellytools/delly" target="_blank" rel="noopener noreferrer">Delly<svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg></a>. <a href="https://www.ncbi.nlm.nih.gov/pubmed/22962449" target="_blank" rel="noopener noreferrer">Delly<svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg></a> calls structural variants jointly on the tumor and normal genome and outputs a <a href="https://samtools.github.io/hts-specs" target="_blank" rel="noopener noreferrer">BCF<svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg></a> file, the binary encoding of <a href="https://samtools.github.io/hts-specs" target="_blank" rel="noopener noreferrer">VCF<svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg></a>. You can also provide a text file with regions to exclude from the analysis of structural variants. The default exclude map of Delly includes the telomeric and centromeric regions of all human chromosomes since these regions cannot be accurately analyzed with short-read data.</p> <div class="language-bash extra-class"><pre class="language-bash"><code><span class="token function">cd</span> /data/sv
delly call -n -q 20 -g chr2.fa -x hg19.ex -o sv.bcf tumor.bam control.bam
</code></pre></div><p>VCF was originally designed for short variants and that's why all SV callers heavily use the INFO fields to encode additional information about the SV such as the structural variant end (INFO:END) and the SV type (INFO:SVTYPE). You can look at the header of the BCF file using grep, '-A 2' includes the first two structural variant records in the file:</p> <div class="language-bash extra-class"><pre class="language-bash"><code>bcftools view sv.bcf <span class="token operator">|</span> <span class="token function">grep</span> <span class="token string">&quot;^#&quot;</span> -A 2
</code></pre></div><p><a href="https://github.com/dellytools/delly" target="_blank" rel="noopener noreferrer">Delly<svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg></a> uses the VCF:INFO fields for structural variant site information, such as how confident the structural variant prediction is and how accurate the breakpoints are. The genotype fields contain the actual sample genotype, its genotype quality and genotype likelihoods and various count fields for the variant and reference supporting reads and spanning pairs. If you browse through the VCF file you will notice that a subset of the Delly structural variant predictions have been refined using split-reads. These precise variants are flagged in the VCF info field with the tag 'PRECISE', all others are listed as 'IMPRECISE'. Please note that this BCF file contains germline and somatic structural variants but also false positives caused by repeat-induced mis-mappings or incomplete reference sequences. <a href="https://github.com/dellytools/svprops" target="_blank" rel="noopener noreferrer">SVprops<svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg></a> is a simple program that converts Delly's BCF output to a tab-delimited SV site list.</p> <div class="language-bash extra-class"><pre class="language-bash"><code>svprops sv.bcf <span class="token operator">|</span> <span class="token function">head</span>
</code></pre></div><p>Every record of the BCF file is converted to one tab-delimited row with all kinds of summary statistics. <a href="https://github.com/dellytools/svprops" target="_blank" rel="noopener noreferrer">SVprops<svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg></a> also provides a 'column-view' listing summary statistics for all samples present in the BCF file.</p> <div class="language-bash extra-class"><pre class="language-bash"><code>sampleprops sv.bcf
</code></pre></div><p>This initial SV calling cannot differentiate somatic and germline structural variants. For instance, the 2 complex variants we looked at before are still present in the Delly output. A proximal duplication causes 2 paired-end signatures (deletion-type and duplication-type):</p> <div class="language-bash extra-class"><pre class="language-bash"><code>bcftools view sv.bcf chr2:18905691-18907969 <span class="token operator">|</span> <span class="token function">awk</span> <span class="token string">'<span class="token variable">$2</span>&gt;=18905691 &amp;&amp; <span class="token variable">$2</span>&lt;=18907969'</span>
</code></pre></div><p><em><strong>Exercises</strong></em></p> <ul><li>What is the fraction of deletions that has been called precisely (at single nucleotide resolution) by Delly?</li> <li>Is the germline inverted duplication present in Delly's output files?</li> <li>What type of SVs delineate a proximal inverted duplication?</li> <li>Some of the SVs have nucleotide resolution and the alternative haplotype is present in INFO:CONSENSUS. <a href="https://genome.ucsc.edu/cgi-bin/hgBlat" target="_blank" rel="noopener noreferrer">Blat<svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg></a> the consensus sequence of some of these deletions. What genomic element has been deleted for DEL00001919 (bcftools view sv.bcf | grep &quot;DEL00001919&quot;)? Is it a known variant?</li></ul> <h2 id="somatic-filtering"><a href="#somatic-filtering" aria-hidden="true" class="header-anchor">#</a> Somatic Filtering</h2> <p>Delly ships with a basic somatic filtering subcommand that uses the matched control and possibly additional control samples from unrelated individuals. The somatic filtering requires a sample file listing tumor and control sample names from the VCF file.</p> <div class="language-bash extra-class"><pre class="language-bash"><code><span class="token function">cat</span> spl.tsv
</code></pre></div><p>There are many parameters available to tune the somatic structural variant filtering. Below we require a minimum variant allele frequency of 25%, no support in the matched normal and an overall confident structural variant site prediction with the VCF filter field being equal to PASS.</p> <div class="language-bash extra-class"><pre class="language-bash"><code>delly filter -p -f somatic -o somatic.bcf -a 0.25 -s spl.tsv sv.bcf
sampleprops somatic.bcf
</code></pre></div><p><em><strong>Exercises</strong></em></p> <ul><li>What is the average size of the somatic SVs?</li> <li>Is there any SV type that is enriched among the somatic SVs?</li></ul> <h2 id="structural-variant-visualization-using-igv"><a href="#structural-variant-visualization-using-igv" aria-hidden="true" class="header-anchor">#</a> Structural Variant Visualization using IGV</h2> <p>IGV is excellent for inspecting small variants but for these long-range complex re-arrangements its less powerful. You cannot view the entire event in IGV but you can still visualize the breakpoint regions separately, denoted as left breakpoint (L) and right breakpoint (R) below.</p> <div class="language-bash extra-class"><pre class="language-bash"><code>svprops somatic.bcf <span class="token operator">|</span> <span class="token function">tail</span> -n +2 <span class="token operator">|</span> <span class="token function">awk</span> <span class="token string">'{print <span class="token variable">$1</span>&quot;\t&quot;(<span class="token variable">$2</span>-500)&quot;\t&quot;(<span class="token variable">$2</span>+500)&quot;\t&quot;<span class="token variable">$5</span>&quot;L&quot;;}'</span> <span class="token operator">&gt;</span> somatic.bp.bed
svprops somatic.bcf <span class="token operator">|</span> <span class="token function">tail</span> -n +2 <span class="token operator">|</span> <span class="token function">awk</span> <span class="token string">'{print <span class="token variable">$3</span>&quot;\t&quot;(<span class="token variable">$4</span>-500)&quot;\t&quot;(<span class="token variable">$4</span>+500)&quot;\t&quot;<span class="token variable">$5</span>&quot;R&quot;;}'</span> <span class="token operator">&gt;&gt;</span> somatic.bp.bed
<span class="token function">sort</span> -k4,4 somatic.bp.bed
</code></pre></div><h2 id="interactive-igv-2"><a href="#interactive-igv-2" aria-hidden="true" class="header-anchor">#</a> Interactive IGV</h2> <p>With an X11 display we can start <a href="http://software.broadinstitute.org/software/igv/" target="_blank" rel="noopener noreferrer">IGV<svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg></a> and browse the variants interactively using the chr2 reference sequence.</p> <div class="language-bash extra-class"><pre class="language-bash"><code>./igv.sh -g chr2.fa
</code></pre></div><p>As previously, once IGV has started use 'File' and 'Load from File' to load the tumor and normal bam alignment file. Then import 'somatic.bp.bed' from your working directory using 'Regions' and 'Import Regions'. The somatic structural variants can then be browsed easily using 'Regions' and 'Region Navigator' and you probably want to switch on 'View as pairs' and 'Color alignments by pair orientation'. You may also want to play around with the split screen view, where you can right click a read and select 'View mate region in split screen'.</p> <h2 id="igv-in-batch-mode-2"><a href="#igv-in-batch-mode-2" aria-hidden="true" class="header-anchor">#</a> IGV in batch mode</h2> <p>Without an X11 display we can still use IGV to plot snapshots of the putative SVs.</p> <div class="language-bash extra-class"><pre class="language-bash"><code>./runIGVBatch.sh somatic.bp.bed
<span class="token comment"># For instance, an inversion (left breakpoint)</span>
<span class="token function">open</span> INV00000238L.png
<span class="token comment"># and the right breakpoint</span>
<span class="token function">open</span> INV00000238R.png
</code></pre></div><h2 id="visualizing-complex-rearrangements"><a href="#visualizing-complex-rearrangements" aria-hidden="true" class="header-anchor">#</a> Visualizing Complex Rearrangements</h2> <p>To reveal any complex re-arrangement patterns it's worthwhile to create a so-called read-depth ratio plot using <a href="https://www.r-project.org" target="_blank" rel="noopener noreferrer">R statistics<svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg></a> and overlay the somatic structural variants. We therefore first calculate fragement counts in non-overlapping 10kbp windows using <a href="https://github.com/tobiasrausch/alfred" target="_blank" rel="noopener noreferrer">Alfred<svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg></a>.</p> <div class="language-bash extra-class"><pre class="language-bash"><code>alfred count_dna -o tumor.cov.gz tumor.bam
zcat tumor.cov.gz <span class="token operator">|</span> <span class="token function">head</span>
alfred count_dna -o control.cov.gz control.bam
zcat control.cov.gz <span class="token operator">|</span> <span class="token function">head</span>
</code></pre></div><p><a href="https://github.com/tobiasrausch/alfred" target="_blank" rel="noopener noreferrer">Alfred<svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg></a> has a simple Rscript to plot these raw fragment counts across the genome. In our case,</p> <div class="language-bash extra-class"><pre class="language-bash"><code>Rscript /opt/alfred/R/rd.R tumor.cov.gz
convert tumor.wholegenome.pdf tumor.wholegenome.png
<span class="token function">open</span> tumor.wholegenome.png
Rscript /opt/alfred/R/rd.R control.cov.gz
convert control.wholegenome.pdf control.wholegenome.png
<span class="token function">open</span> control.wholegenome.png
</code></pre></div><p>For a matched tumor-control sample it is common practice to normalize the tumor against the control (log2 read-depth ratio plot). This method is implement in the Rscript rdbaf.R.</p> <div class="language-bash extra-class"><pre class="language-bash"><code><span class="token function">paste</span> <span class="token operator">&lt;</span><span class="token punctuation">(</span>zcat tumor.cov.gz<span class="token punctuation">)</span> <span class="token operator">&lt;</span><span class="token punctuation">(</span>zcat control.cov.gz <span class="token operator">|</span> <span class="token function">cut</span> -f 5<span class="token punctuation">)</span> <span class="token operator">|</span> <span class="token function">gzip</span> -c <span class="token operator">&gt;</span> cov.gz
Rscript rdbaf.R cov.gz
<span class="token function">open</span> cov.png
</code></pre></div><p>Next, we will try to visualize the somatic SV calls on top of the log2 read-depth ratio plot.</p> <div class="language-bash extra-class"><pre class="language-bash"><code>svprops somatic.bcf <span class="token operator">&gt;</span> svs.tsv
Rscript rdbaf.R cov.gz svs.tsv
<span class="token function">open</span> cov.png
</code></pre></div><h2 id="allelic-depth-for-snps"><a href="#allelic-depth-for-snps" aria-hidden="true" class="header-anchor">#</a> Allelic Depth for SNPs</h2> <p>We can also augment the plots with SNP variant allele frequencies. The file 'snps.pos' contains 1000 Genomes SNP locations on chr2 where the control sample is heterozygous.</p> <div class="language-bash extra-class"><pre class="language-bash"><code><span class="token function">head</span> snps.pos
</code></pre></div><p>Using <a href="http://www.htslib.org" target="_blank" rel="noopener noreferrer">SAMtools<svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg></a> and <a href="http://www.htslib.org" target="_blank" rel="noopener noreferrer">BCFtools<svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg></a> we can annotate the allelic depth for every SNP.</p> <div class="language-bash extra-class"><pre class="language-bash"><code>samtools mpileup -go mp.bcf -f chr2.fa -t AD -l snps.pos tumor.bam
bcftools call -vmAO z -o mp.vcf.gz mp.bcf
zcat mp.vcf.gz <span class="token operator">|</span> <span class="token function">grep</span> <span class="token string">&quot;^#&quot;</span> -A 1
</code></pre></div><p>Let us first filter for bi-allelic SNPs at coverage &gt;=5 and then extract AD (allelic depth). In the last step we calculate the so-called B-allele frequency.</p> <div class="language-bash extra-class"><pre class="language-bash"><code>bcftools filter -i <span class="token string">'DP&gt;=10'</span> mp.vcf.gz <span class="token operator">|</span> bcftools view -m2 -M2 -v snps - <span class="token operator">|</span> bcftools query -f <span class="token string">'%CHROM\t%POS[\t%AD]\n'</span> <span class="token operator">&gt;</span> ad.tsv
<span class="token function">cat</span> ad.tsv <span class="token operator">|</span> <span class="token function">tr</span> <span class="token string">','</span> <span class="token string">'\t'</span> <span class="token operator">|</span> <span class="token function">awk</span> <span class="token string">'{print <span class="token variable">$2</span>&quot;\t&quot;(<span class="token variable">$4</span>)/(<span class="token variable">$3</span>+<span class="token variable">$4</span>);}'</span> <span class="token operator">&gt;</span> baf.tsv
</code></pre></div><p>Next, we integrate the B-allele frequency plot with the read-depth plot and the structural variants.</p> <div class="language-bash extra-class"><pre class="language-bash"><code>Rscript rdbaf.R cov.gz svs.tsv baf.tsv
<span class="token function">open</span> cov.png
</code></pre></div></div> <footer class="page-edit"><!----> <!----></footer> <div class="page-nav"><p class="inner"><span class="prev">
        ←
        <a href="/courses/vc/snv/" class="prev">
          SNV Calling
        </a></span> <span class="next"><a href="/courses/vc/appendix/">
          Appendix
        </a>
        →
      </span></p></div> </main></div><div class="global-ui"><!----></div></div>
    <script src="/courses/vc/assets/js/app.eb2d6309.js" defer></script><script src="/courses/vc/assets/js/2.65bda2c6.js" defer></script><script src="/courses/vc/assets/js/9.119cb8a5.js" defer></script>
  </body>
</html>
