---
title: Projects
layout: default
permalink: "/projects/"
---

## Selected Publications ([Google Scholar profile](https://scholar.google.de/citations?user=fQ1VoZEAAAAJ))

[![DOI:10.1038/nature15394](https://zenodo.org/badge/DOI/10.1038/nature15394.svg)](https://doi.org/10.1038/nature15394)

[1000 Genomes Structural Variant Calling](https://www.nature.com/articles/nature15394): A catalogue of human genetic structural variation.

---

[![DOI:10.1038/s41467-018-08148-z](https://zenodo.org/badge/DOI/10.1038/s41467-018-08148-z.svg)](https://doi.org/10.1038/s41467-018-08148-z)

[Haplotype-resolved Stuctural Variant Calling](https://www.nature.com/articles/s41467-018-08148-z): Multi-platform discovery of haplotype-resolved structural variation in human genomes.

---

[![DOI:10.1126/science.abf7117](https://zenodo.org/badge/DOI/10.1126/science.abf7117.svg)](https://doi.org/10.1126/science.abf7117)

[Haplotype-resolved Human Genomes](https://doi.org/10.1126/science.abf7117): Haplotype-resolved diverse human genomes and integrated analysis of structural variation.

---

## Cancer Genomics Publications

[![DOI:10.1016/j.xgen.2023.100281](https://zenodo.org/badge/DOI/10.1016/j.xgen.2023.100281.svg)](https://doi.org/10.1016/j.xgen.2023.100281)

[Long-read sequencing of a cancer genome](https://doi.org/10.1016/j.xgen.2023.100281): Long-read sequencing of diagnosis and post-therapy medulloblastoma reveals complex rearrangement patterns and epigenetic signatures.

---

[![DOI:10.1016/j.cell.2011.12.013](https://zenodo.org/badge/DOI/10.1016/j.cell.2011.12.013.svg)](https://doi.org/10.1016/j.cell.2011.12.013)

[Chromothripsis in Medulloblastoma](https://www.sciencedirect.com/science/article/pii/S0092867411015169): Genome sequencing of pediatric medulloblastoma samples to understand complex, somatic DNA rearrangements.

---

[![DOI:10.3324/haematol.2015.129692](https://zenodo.org/badge/DOI/10.3324/haematol.2015.12969.svg)](https://doi.org/10.3324/haematol.2015.129692)

[Relapses of T-cell acute lymphoblastic leukemia](http://www.haematologica.org/content/early/2015/08/17/haematol.2015.129692): Pediatric T-cell acute lymphoblastic leukemia evolves into relapse by clonal selection and acquisition of mutations.

---

[![DOI:10.3324/haematol.2014.104992](https://zenodo.org/badge/DOI/10.3324/haematol.2014.104992.svg)](https://doi.org/10.3324/haematol.2014.104992)

[Activating STAT5B N642H mutations in T-cell acute lymphoblastic leukemia](http://www.haematologica.org/content/99/10/e188.long): The activating STAT5B N642H mutation is a common abnormality in pediatric T-cell acute lymphoblastic leukemia and confers a higher risk of relapse.

---

[![DOI:10.1038/s41586-019-1815-x](https://zenodo.org/badge/DOI/10.1038/s41586-019-1815-x.svg)](https://doi.org/10.1038/s41586-019-1815-x)

[The molecular landscape of ETMR at diagnosis and relapse](https://www.nature.com/articles/s41586-019-1815-x): ETMR tumours in which the proposed driver C19MC2 was not amplified frequently had germline mutations in DICER1 or other microRNA-related aberrations.

---

[![DOI:10.1038/s41586-020-1969-6](https://zenodo.org/badge/DOI/10.1038/s41586-020-1969-6.svg)](https://doi.org/10.1038/s41586-020-1969-6)

[Pan-Cancer Analysis](https://doi.org/10.1038/s41586-020-1969-6): Germline determinants of the somatic mutation landscape in 2,642 cancer genomes.

---


## Rare Disease Publications

[![DOI:10.1016/j.jaci.2013.02.012](https://zenodo.org/badge/DOI/10.1016/j.jaci.2013.02.012.svg)](https://doi.org/10.1016/j.jaci.2013.02.012)

[Severe combined immunodeficiency](https://www.sciencedirect.com/science/article/pii/S0091674913003217): CARD11 inactivation leads to severe combined immunodeficiency.

---

[![DOI:10.1182/blood.2021013519](https://zenodo.org/badge/DOI/10.1182/blood.2021013519.svg)](https://doi.org/10.1182/blood.2021013519)

[Novel subtype of hemochromatosis](https://doi.org/10.1182/blood.2021013519): Constitutional PIGA mutations cause a novel subtype of hemochromatosis in patients with neurologic dysfunction

---

## Web Applications for Molecular Biologists

[![DOI:10.1186/s12864-020-6635-8](https://zenodo.org/badge/DOI/10.1186/s12864-020-6635-8.svg)](https://doi.org/10.1186/s12864-020-6635-8)

[www.gear-genomics.com](https://www.gear-genomics.com/): Genome Analysis Server with interactive and efficient web applications for molecular biologists.

---

## Computational Biology Methods

[![DOI:10.1093/bioinformatics/bts378](https://zenodo.org/badge/DOI/10.1093/bioinformatics/bts378.svg)](https://doi.org/10.1093/bioinformatics/bts378)

[Delly](https://github.com/dellytools/delly/): Structural variant discovery by integrated paired-end and split-read analysis.


---

[![DOI:10.1093/bioinformatics/bty1007](https://zenodo.org/badge/DOI/10.1093/bioinformatics/bty1007.svg)](https://doi.org/10.1093/bioinformatics/bty1007)

[Alfred](https://github.com/tobiasrausch/alfred/): BAM alignment statistics, feature counting and feature annotation.

---

[![DOI:10.1186/s12864-020-6635-8](https://zenodo.org/badge/DOI/10.1186/s12864-020-6635-8.svg)](https://doi.org/10.1186/s12864-020-6635-8)

[Tracy](https://github.com/gear-genomics/tracy/): Basecalling, alignment and deconvolution of Sanger Chromatogram trace files.

---

[![DOI:10.1016/j.xgen.2023.100281](https://zenodo.org/badge/DOI/10.1016/j.xgen.2023.100281.svg)](https://doi.org/10.1016/j.xgen.2023.100281)

[Wally](https://github.com/tobiasrausch/wally/): Visualization of aligned sequencing reads and contigs.

---

[![DOI:10.1093/bioinformatics/btz719](https://zenodo.org/badge/DOI/10.1093/bioinformatics/btz719.svg)](https://doi.org/10.1093/bioinformatics/btz719)

[Visor](https://github.com/davidebolo1993/VISOR/): Variant simulator for short, long and linked reads

---

## SeqAn Contributions

[![DOI:10.1186/1471-2105-9-11](https://zenodo.org/badge/DOI/10.1186/1471-2105-9-11.svg)](https://doi.org/10.1186/1471-2105-9-11)

[SeqAn](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-11): An efficient, generic C++ library for sequence analysis.

---

[![DOI:10.1093/bioinformatics/btn281](https://zenodo.org/badge/DOI/10.1093/bioinformatics/btn281.svg)](https://doi.org/10.1093/bioinformatics/btn281)

[Multiple Sequence Alignment](https://academic.oup.com/bioinformatics/article-lookup/doi/10.1093/bioinformatics/btn281): Segment-based multiple sequence alignment using SeqAn.

---

## Publications

A full list of co-authored publications is available on [Google Scholar](https://scholar.google.de/citations?user=fQ1VoZEAAAAJ).

Publications and peer review statistics are also available on [ORCID](https://orcid.org/0000-0001-5773-5620).
