<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">


<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>Structural Variant Analysis using Simulated Data &mdash; ngs2016 January 2016 documentation</title>
    
    <link rel="stylesheet" href="_static/default.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    './',
        VERSION:     'January 2016',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <link rel="top" title="ngs2016 January 2016 documentation" href="index.html" />
    <link rel="next" title="References" href="citations.html" />
    <link rel="prev" title="Cancer Genomics 2016" href="index.html" /> 
  </head>
  <body>
    <div class="related">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="right" >
          <a href="citations.html" title="References"
             accesskey="N">next</a> |</li>
        <li class="right" >
          <a href="index.html" title="Cancer Genomics 2016"
             accesskey="P">previous</a> |</li>
        <li><a href="index.html">ngs2016 January 2016 documentation</a> &raquo;</li> 
      </ul>
    </div>  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body">
            
  <div class="section" id="structural-variant-analysis-using-simulated-data">
<h1>Structural Variant Analysis using Simulated Data<a class="headerlink" href="#structural-variant-analysis-using-simulated-data" title="Permalink to this headline">¶</a></h1>
<div class="section" id="getting-started">
<h2>Getting Started<a class="headerlink" href="#getting-started" title="Permalink to this headline">¶</a></h2>
<p>As a quick introduction we will do a structural variant analysis of the simulated data provided by the ICGC-TCGA DREAM Mutation Calling challenge <a class="reference internal" href="citations.html#dream" id="id1">[Dream]</a>. The data has been further down-sampled to speed up all analyses discussed hereafter. The entire data can be downloaded from Google Drive using this URL &#8216;<a class="reference external" href="https://drive.google.com/file/d/0BzeE4HZ6nt-KMDV2d0s2N1pZQkE/view?usp=sharing">https://drive.google.com/file/d/0BzeE4HZ6nt-KMDV2d0s2N1pZQkE/view?usp=sharing</a>&#8216;.</p>
</div>
<div class="section" id="alignment">
<h2>Alignment<a class="headerlink" href="#alignment" title="Permalink to this headline">¶</a></h2>
<p>The simulated data has already been aligned to chr20 of hg19. The chr20 reference FASTA file is called &#8216;20.fa&#8217;. There is one bam file <a class="reference internal" href="citations.html#bam" id="id2">[Bam]</a> for the tumor (file tumor20.bam) and one for the matched normal (file normal20.bam). Using samtools <a class="reference internal" href="citations.html#li2009" id="id3">[Li2009]</a> we can have a look at the first few records.</p>
<div class="highlight-bash"><div class="highlight"><pre>samtools view <span class="nv">$DS</span>/tumor20.bam | head
</pre></div>
</div>
<p>If you haven&#8217;t done so yet please familiarize yourself with the bam format, the required fields present in every bam alignment record are explained below:</p>
<blockquote>
<div><table border="1" class="docutils">
<colgroup>
<col width="13%" />
<col width="19%" />
<col width="69%" />
</colgroup>
<tbody valign="top">
<tr class="row-odd"><td><strong>Col</strong></td>
<td><strong>Field</strong></td>
<td><strong>Description</strong></td>
</tr>
<tr class="row-even"><td>1</td>
<td>QNAME</td>
<td>Query template NAME</td>
</tr>
<tr class="row-odd"><td>2</td>
<td>FLAG</td>
<td>bitwise FLAG</td>
</tr>
<tr class="row-even"><td>3</td>
<td>RNAME</td>
<td>Reference sequence NAME</td>
</tr>
<tr class="row-odd"><td>4</td>
<td>POS</td>
<td>1-based leftmost mapping POSition</td>
</tr>
<tr class="row-even"><td>5</td>
<td>MAPQ</td>
<td>MAPping Quality</td>
</tr>
<tr class="row-odd"><td>6</td>
<td>CIGAR</td>
<td>CIGAR string</td>
</tr>
<tr class="row-even"><td>7</td>
<td>RNEXT</td>
<td>Ref. name of the mate/next read</td>
</tr>
<tr class="row-odd"><td>8</td>
<td>PNEXT</td>
<td>Position of the mate/next read</td>
</tr>
<tr class="row-even"><td>9</td>
<td>TLEN</td>
<td>observed Template LENgth</td>
</tr>
<tr class="row-odd"><td>10</td>
<td>SEQ</td>
<td>segment SEQuence</td>
</tr>
<tr class="row-even"><td>11</td>
<td>QUAL</td>
<td>ASCII of Phred-scaled base QUALity+33</td>
</tr>
</tbody>
</table>
</div></blockquote>
<p>The bitwise FLAG can be decoded using the explain flag tool from the picard distribution <a class="reference internal" href="citations.html#explainflag" id="id4">[ExplainFlag]</a>.</p>
<p>Usually one would first evaluate the quality of the sequencing data using tools such as FASTQC <a class="reference internal" href="citations.html#fastqc" id="id5">[FastQC]</a> but since this is simulated data we will skip this step. Other useful tools for NGS quality control are picard <a class="reference internal" href="citations.html#picard" id="id6">[Picard]</a> and samtools <a class="reference internal" href="citations.html#samtools" id="id7">[Samtools]</a>.</p>
<p>Delly <a class="reference internal" href="citations.html#rausch2012" id="id8">[Rausch2012]</a> uses paired-end mapping to discover structural variants and the method can handle paired-ends from different libraries as long as these have been tagged with unique read-groups (&#64;RG tags). These read-groups are listed in the bam header.</p>
<div class="highlight-bash"><div class="highlight"><pre>samtools view -H <span class="nv">$DS</span>/tumor20.bam
</pre></div>
</div>
</div>
<div class="section" id="structural-variant-discovery-using-delly">
<h2>Structural Variant Discovery using Delly<a class="headerlink" href="#structural-variant-discovery-using-delly" title="Permalink to this headline">¶</a></h2>
<p>Since this is simulated data we will skip the alignment quality control and move on directly to the structural variant calling. Delly <a class="reference internal" href="citations.html#rausch2012" id="id9">[Rausch2012]</a> calls structural variants jointly on the tumor and normal genome and outputs a VCF (variant-call-format) file. The VCF specification can be found on the VCFtools website <a class="reference internal" href="citations.html#vcftools" id="id10">[VCFtools]</a>. The ICGC-TCGA DREAM Mutation Calling challenge simulated only deletions, duplications and inversions so we will skip the translocation prediction. Besides the bam files, you can provide a reference genome which then triggers the split-read search of Delly. You can also provide a text file with regions to exclude from the analysis of structural variants. The default exclude map of Delly includes the telomeric and centromeric regions of all human chromosomes since these regions cannot be accurately analyzed with short-read data.</p>
<div class="highlight-bash"><div class="highlight"><pre><span class="nv">$DE</span>/src/delly -t DEL -g <span class="nv">$DS</span>/20.fa -x <span class="nv">$DS</span>/hg19.ex -o del.vcf <span class="nv">$DS</span>/tumor20.bam <span class="nv">$DS</span>/normal20.bam
<span class="nv">$DE</span>/src/delly -t DUP -g <span class="nv">$DS</span>/20.fa -x <span class="nv">$DS</span>/hg19.ex -o dup.vcf <span class="nv">$DS</span>/tumor20.bam <span class="nv">$DS</span>/normal20.bam
<span class="nv">$DE</span>/src/delly -t INV -g <span class="nv">$DS</span>/20.fa -x <span class="nv">$DS</span>/hg19.ex -o inv.vcf <span class="nv">$DS</span>/tumor20.bam <span class="nv">$DS</span>/normal20.bam
</pre></div>
</div>
<p>A vcf file <a class="reference internal" href="citations.html#vcftools" id="id11">[VCFtools]</a> has multiple header lines starting with the hash # sign. Below the header lines is one record for each structural variant. The record format is described in the below table:</p>
<blockquote>
<div><table border="1" class="docutils">
<colgroup>
<col width="7%" />
<col width="10%" />
<col width="83%" />
</colgroup>
<tbody valign="top">
<tr class="row-odd"><td><strong>Col</strong></td>
<td><strong>Field</strong></td>
<td><strong>Description</strong></td>
</tr>
<tr class="row-even"><td>1</td>
<td>CHROM</td>
<td>Chromosome name</td>
</tr>
<tr class="row-odd"><td>2</td>
<td>POS</td>
<td>1-based position. For an indel, this is the position preceding the indel.</td>
</tr>
<tr class="row-even"><td>3</td>
<td>ID</td>
<td>Variant identifier. Usually the dbSNP rsID.</td>
</tr>
<tr class="row-odd"><td>4</td>
<td>REF</td>
<td>Reference sequence at POS involved in the variant. For a SNP, it is a single base.</td>
</tr>
<tr class="row-even"><td>5</td>
<td>ALT</td>
<td>Comma delimited list of alternative sequence(s).</td>
</tr>
<tr class="row-odd"><td>6</td>
<td>QUAL</td>
<td>Phred-scaled probability of all samples being homozygous reference.</td>
</tr>
<tr class="row-even"><td>7</td>
<td>FILTER</td>
<td>Semicolon delimited list of filters that the variant fails to pass.</td>
</tr>
<tr class="row-odd"><td>8</td>
<td>INFO</td>
<td>Semicolon delimited list of variant information.</td>
</tr>
<tr class="row-even"><td>9</td>
<td>FORMAT</td>
<td>Colon delimited list of the format of individual genotypes in the following fields.</td>
</tr>
<tr class="row-odd"><td>10+</td>
<td>Samples</td>
<td>Individual genotype information defined by FORMAT.</td>
</tr>
</tbody>
</table>
</div></blockquote>
<p>You can look at the header of the vcf file using grep, &#8216;-A 1&#8217; includes the first structural variant record in the file:</p>
<div class="highlight-bash"><div class="highlight"><pre>grep <span class="s2">&quot;^#&quot;</span> -A 1 inv.vcf
</pre></div>
</div>
<p>In general, Delly uses the VCF:INFO fields for structural variant site information, such as how confident the structural variant prediction is and how accurate the breakpoints are. The genotype fields contain the actual sample genotype, its genotype quality and genotype likelihoods and various count fields for the variant and reference supporting reads and spanning pairs. If you browse through the vcf file you will notice that a subset of the Delly structural variant predictions have been refined using split-reads. These precise variants are flagged in the vcf info field with the tag &#8216;PRECISE&#8217;. To count the number of precise and imprecise variants you can simply use grep.</p>
<div class="highlight-bash"><div class="highlight"><pre>grep -c -w <span class="s2">&quot;PRECISE&quot;</span> *.vcf
grep -c -w <span class="s2">&quot;IMPRECISE&quot;</span> *.vcf
</pre></div>
</div>
<p>Please note that this vcf file contains germline and somatic structural variants but also false positives caused by repeat-induced mis-mappings or incomplete reference sequences. As a final step we have to use the structural variant site information and the tumor and normal genotype information to filter a set of confident somatic structural variants.</p>
</div>
<div class="section" id="somatic-structural-variant-filtering">
<h2>Somatic Structural Variant Filtering<a class="headerlink" href="#somatic-structural-variant-filtering" title="Permalink to this headline">¶</a></h2>
<p>Delly ships with a somatic filtering python script. For a set of confident somatic calls one could exclude all structural variants &lt;400bp, require a minimum variant allele frequency of 10%, no support in the matched normal and an overall confident structural variant site prediction with the VCF filter field being equal to PASS.</p>
<div class="highlight-bash"><div class="highlight"><pre>python <span class="nv">$SF</span> -t DEL -v del.vcf -o del.filt.vcf -a 0.1 -m 400 -f
python <span class="nv">$SF</span> -t DUP -v dup.vcf -o dup.filt.vcf -a 0.1 -m 400 -f
python <span class="nv">$SF</span> -t INV -v inv.vcf -o inv.filt.vcf -a 0.1 -m 400 -f
</pre></div>
</div>
<p>Using VCFtools <a class="reference internal" href="citations.html#vcftools" id="id12">[VCFtools]</a> we can merge all somatic structural variants together in a single vcf file.</p>
<div class="highlight-bash"><div class="highlight"><pre>vcf-concat del.filt.vcf dup.filt.vcf inv.filt.vcf  | vcf-sort &gt; somatic.sv.vcf
vcf-validator somatic.sv.vcf
</pre></div>
</div>
<p>For large VCF files you should also zip and index them using bgzip and tabix.</p>
<div class="highlight-bash"><div class="highlight"><pre>bgzip somatic.sv.vcf
tabix somatic.sv.vcf.gz
</pre></div>
</div>
<p>The final step will be to browse some of these somatic structural variants in IGV <a class="reference internal" href="citations.html#igv" id="id13">[IGV]</a>. To make that easier we will also create a bed file with the start and end coordinates of each structural variant. This vcf2bed conversion is a bit ugly because VCF was designed for single-nucleotide variants, so I wrote a small shell script to do this conversion, called &#8216;vcf2Bed.sh&#8217;.</p>
<div class="highlight-bash"><div class="highlight"><pre><span class="nv">$DE</span>/converter/dellyVcf2Bed.sh somatic.sv.vcf.gz &gt; somatic.sv.bed
head somatic.sv.bed
</pre></div>
</div>
<p>We then start IGV using the chr20 reference.</p>
<div class="highlight-bash"><div class="highlight"><pre>igv.sh -g <span class="nv">$DS</span>/20.fa
</pre></div>
</div>
<p>Once IGV has started use &#8216;File&#8217; and &#8216;Load from File&#8217; to load the tumor and normal bam alignment file. Then import &#8216;somatic.sv.bed&#8217; from your working directory using &#8216;Regions&#8217; and &#8216;Import Regions&#8217;. The somatic structural variants can then be browsed easily using &#8216;Regions&#8217; and &#8216;Region Navigator&#8217;. Select a structural variant in the Region Navigator and click &#8216;View&#8217;, which will center the IGV alignment view on the selected structural variant. It&#8217;s usually best to zoom out once then by clicking on the &#8216;-&#8216; sign in the toolbar at the top, so you can view all supporting abnormal paired-ends. To highlight the abnormal paired-ends please right click in IGV on tumor20.bam and activate &#8216;View as pairs&#8217;. In the same menu, please open &#8216;Color alignments by&#8217; and then switch to &#8220;pair orientation&#8217;, afterwards repeat these steps for normal20.bam. For deletions, you can also color the alignments by &#8220;insert size&#8221;. You can obviously play around with all the different visualization options but the above two are what I found most useful for a discordant paired-end analysis. Using the Region Navigator you can easily iterate now through the predicted, somatic SVs.</p>
</div>
</div>


          </div>
        </div>
      </div>
      <div class="sphinxsidebar">
        <div class="sphinxsidebarwrapper">
  <h3><a href="index.html">Table Of Contents</a></h3>
  <ul>
<li><a class="reference internal" href="#">Structural Variant Analysis using Simulated Data</a><ul>
<li><a class="reference internal" href="#getting-started">Getting Started</a></li>
<li><a class="reference internal" href="#alignment">Alignment</a></li>
<li><a class="reference internal" href="#structural-variant-discovery-using-delly">Structural Variant Discovery using Delly</a></li>
<li><a class="reference internal" href="#somatic-structural-variant-filtering">Somatic Structural Variant Filtering</a></li>
</ul>
</li>
</ul>

  <h4>Previous topic</h4>
  <p class="topless"><a href="index.html"
                        title="previous chapter">Cancer Genomics 2016</a></p>
  <h4>Next topic</h4>
  <p class="topless"><a href="citations.html"
                        title="next chapter">References</a></p>
  <h3>This Page</h3>
  <ul class="this-page-menu">
    <li><a href="_sources/chapter01.txt"
           rel="nofollow">Show Source</a></li>
  </ul>
<div id="searchbox" style="display: none">
  <h3>Quick search</h3>
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" />
      <input type="submit" value="Go" />
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
    <p class="searchtip" style="font-size: 90%">
    Enter search terms or a module, class or function name.
    </p>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             >index</a></li>
        <li class="right" >
          <a href="citations.html" title="References"
             >next</a> |</li>
        <li class="right" >
          <a href="index.html" title="Cancer Genomics 2016"
             >previous</a> |</li>
        <li><a href="index.html">ngs2016 January 2016 documentation</a> &raquo;</li> 
      </ul>
    </div>
    <div class="footer">
        &copy; Copyright 2016, Tobias Rausch.
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 1.2b1.
    </div>
  </body>
</html>