<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">


<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>Alignment &amp; Data Quality Control &mdash; ngs2014 Oct 2014 documentation</title>
    
    <link rel="stylesheet" href="_static/default.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    './',
        VERSION:     'Oct 2014',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <link rel="top" title="ngs2014 Oct 2014 documentation" href="index.html" />
    <link rel="next" title="DNA Variant Calling" href="chapter02.html" />
    <link rel="prev" title="Welcome to NGS 2014’s documentation!" href="index.html" /> 
  </head>
  <body>
    <div class="related">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="right" >
          <a href="chapter02.html" title="DNA Variant Calling"
             accesskey="N">next</a> |</li>
        <li class="right" >
          <a href="index.html" title="Welcome to NGS 2014’s documentation!"
             accesskey="P">previous</a> |</li>
        <li><a href="index.html">ngs2014 Oct 2014 documentation</a> &raquo;</li> 
      </ul>
    </div>  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body">
            
  <div class="section" id="alignment-data-quality-control">
<h1>Alignment &amp; Data Quality Control<a class="headerlink" href="#alignment-data-quality-control" title="Permalink to this headline">¶</a></h1>
<div class="section" id="getting-started">
<h2>Getting Started<a class="headerlink" href="#getting-started" title="Permalink to this headline">¶</a></h2>
<p>The course data is from a rare disease sample that was recently sequenced at EMBL Genecore using a DNA whole-exome capture protocol and an illumina sequencer.
The data has been down-sampled and reduced to a single chromosome (chr7) to speed up all analyses discussed hereafter. Please do not copy or redistribute the data anywhere
outside this course. Since all course participants will do the exercises simultaneously please copy the data to your local hard drive using:</p>
<div class="highlight-bash"><div class="highlight"><pre>mkdir &lt;your_name&gt;
<span class="nb">cd</span> &lt;your_name&gt;
tar -xzf /EMBL/scrap/rausch/ngs2014.tar.gz
<span class="nb">cd </span>ngs2014/
</pre></div>
</div>
<p>For VCFtools <a class="reference internal" href="citations.html#vcftools" id="id1">[VCFtools]</a> we also need to set the Perl library path.</p>
<div class="highlight-bash"><div class="highlight"><pre><span class="nb">export </span><span class="nv">PERL5LIB</span><span class="o">=</span><span class="sb">`</span><span class="nb">pwd</span><span class="sb">`</span>/bin/vcftools_0.1.12a/perl/
<span class="nb">export </span><span class="nv">PI</span><span class="o">=</span>/g/teachinglab/bin/
</pre></div>
</div>
<p>If you encounter any errors or typos in this handout please write me an email <a class="reference internal" href="citations.html#rausch" id="id2">[Rausch]</a>.</p>
</div>
<div class="section" id="fastq-qc">
<h2>Fastq QC<a class="headerlink" href="#fastq-qc" title="Permalink to this headline">¶</a></h2>
<p>Before starting to map the data one should inspect the quality of the reads in fastq format <a class="reference internal" href="citations.html#fastq" id="id3">[Fastq]</a> using tools such as fastqc <a class="reference internal" href="citations.html#fastqc" id="id4">[Fastqc]</a>.
Fastqc highlights adapter contamination, PCR duplicates and problems with the base distribution and base qualities.</p>
<div class="highlight-bash"><div class="highlight"><pre>fastqc --noextract r1.fq.gz
unzip r1_fastqc.zip
firefox r1_fastqc/fastqc_report.html

fastqc --noextract r2.fq.gz
unzip r2_fastqc.zip
firefox r2_fastqc/fastqc_report.html
</pre></div>
</div>
<p>If some of the plots are unclear please have a look at the Fastqc documentation <a class="reference internal" href="citations.html#fastqcdoc" id="id5">[FastqcDoc]</a>.</p>
</div>
<div class="section" id="alignment">
<h2>Alignment<a class="headerlink" href="#alignment" title="Permalink to this headline">¶</a></h2>
<p>BWA <a class="reference internal" href="citations.html#li2009a" id="id6">[Li2009a]</a> is a commonly used read mapper. The reference has to be indexed once before it can be used for read alignment:</p>
<div class="highlight-bash"><div class="highlight"><pre>bwa index -a bwtsw chr7.fa
</pre></div>
</div>
<p>Next the paired-end fastq files can be aligned against the index:</p>
<div class="highlight-bash"><div class="highlight"><pre>bwa mem chr7.fa r1.fq.gz r2.fq.gz | samtools view -bT chr7.fa - &gt; align.bam
</pre></div>
</div>
<p>The alignment will run for about 5 minutes. While the alignment is running you can already have a look at the first records of the bam file <a class="reference internal" href="citations.html#bam" id="id7">[Bam]</a> and familiarize yourself with the bam format:</p>
<div class="highlight-bash"><div class="highlight"><pre>samtools view align.bam | head
</pre></div>
</div>
<p>The different required fields present in a bam alignment record are explained below:</p>
<blockquote>
<div><table border="1" class="docutils">
<colgroup>
<col width="13%" />
<col width="19%" />
<col width="69%" />
</colgroup>
<tbody valign="top">
<tr class="row-odd"><td><strong>Col</strong></td>
<td><strong>Field</strong></td>
<td><strong>Description</strong></td>
</tr>
<tr class="row-even"><td>1</td>
<td>QNAME</td>
<td>Query template NAME</td>
</tr>
<tr class="row-odd"><td>2</td>
<td>FLAG</td>
<td>bitwise FLAG</td>
</tr>
<tr class="row-even"><td>3</td>
<td>RNAME</td>
<td>Reference sequence NAME</td>
</tr>
<tr class="row-odd"><td>4</td>
<td>POS</td>
<td>1-based leftmost mapping POSition</td>
</tr>
<tr class="row-even"><td>5</td>
<td>MAPQ</td>
<td>MAPping Quality</td>
</tr>
<tr class="row-odd"><td>6</td>
<td>CIGAR</td>
<td>CIGAR string</td>
</tr>
<tr class="row-even"><td>7</td>
<td>RNEXT</td>
<td>Ref. name of the mate/next read</td>
</tr>
<tr class="row-odd"><td>8</td>
<td>PNEXT</td>
<td>Position of the mate/next read</td>
</tr>
<tr class="row-even"><td>9</td>
<td>TLEN</td>
<td>observed Template LENgth</td>
</tr>
<tr class="row-odd"><td>10</td>
<td>SEQ</td>
<td>segment SEQuence</td>
</tr>
<tr class="row-even"><td>11</td>
<td>QUAL</td>
<td>ASCII of Phred-scaled base QUALity+33</td>
</tr>
</tbody>
</table>
</div></blockquote>
<p>Once the alignment is finished please sort the bam file by coordinate using samtools <a class="reference internal" href="citations.html#li2009b" id="id8">[Li2009b]</a>:</p>
<div class="highlight-bash"><div class="highlight"><pre>samtools sort align.bam align.sort
</pre></div>
</div>
<p>The sorted file now needs to be indexed so that alignment viewers can efficiently access a certain alignment window.</p>
<div class="highlight-bash"><div class="highlight"><pre>samtools index align.sort.bam
</pre></div>
</div>
</div>
<div class="section" id="alignment-qc">
<h2>Alignment QC<a class="headerlink" href="#alignment-qc" title="Permalink to this headline">¶</a></h2>
<p>The quality of the final alignment should be checked prior to any variant calling. There are some simple statistics such as the insert size distribution, the gc bias and the percentage of mapped reads, singletons, duplicates and properly paired reads that can highlight quite a number of typical sequencing problems. Overall it&#8217;s usually worth spending the extra time running these quality control tools. The picard <a class="reference internal" href="citations.html#picard" id="id9">[Picard]</a> and samtools <a class="reference internal" href="citations.html#samtools" id="id10">[Samtools]</a> library provide some excellent tools to compute these simple alignment statistics.</p>
<div class="highlight-bash"><div class="highlight"><pre>samtools flagstat align.sort.bam
java -jar <span class="k">${</span><span class="nv">PI</span><span class="k">}</span>/CollectGcBiasMetrics.jar <span class="nv">I</span><span class="o">=</span>align.sort.bam <span class="nv">O</span><span class="o">=</span>gc.txt <span class="nv">R</span><span class="o">=</span>chr7.fa <span class="nv">CHART</span><span class="o">=</span>gc.pdf <span class="nv">AS</span><span class="o">=</span><span class="nb">true</span>
evince gc.pdf
</pre></div>
</div>
<p>Please read the on-line documentation for these tools to get an explanation of the different metrics <a class="reference internal" href="citations.html#picarddoc" id="id11">[PicardDoc]</a>. There are some general rules how a good alignment should look like, e.g. a heavy gc bias, mapping percentages below 70% or very high duplicate rates are clearly warning signals. However, these statistics vary largely by protocol and hence, it&#8217;s usually best to compare multiple different sequencing runs using the same protocol (DNA-seq, RNA-seq, ChIP-seq, paired-end, single-end or mate-pair protocol, etc.) against each other, which then highlights the outliers.</p>
</div>
</div>


          </div>
        </div>
      </div>
      <div class="sphinxsidebar">
        <div class="sphinxsidebarwrapper">
  <h3><a href="index.html">Table Of Contents</a></h3>
  <ul>
<li><a class="reference internal" href="#">Alignment &amp; Data Quality Control</a><ul>
<li><a class="reference internal" href="#getting-started">Getting Started</a></li>
<li><a class="reference internal" href="#fastq-qc">Fastq QC</a></li>
<li><a class="reference internal" href="#alignment">Alignment</a></li>
<li><a class="reference internal" href="#alignment-qc">Alignment QC</a></li>
</ul>
</li>
</ul>

  <h4>Previous topic</h4>
  <p class="topless"><a href="index.html"
                        title="previous chapter">Welcome to NGS 2014&#8217;s documentation!</a></p>
  <h4>Next topic</h4>
  <p class="topless"><a href="chapter02.html"
                        title="next chapter">DNA Variant Calling</a></p>
  <h3>This Page</h3>
  <ul class="this-page-menu">
    <li><a href="_sources/chapter01.txt"
           rel="nofollow">Show Source</a></li>
  </ul>
<div id="searchbox" style="display: none">
  <h3>Quick search</h3>
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" />
      <input type="submit" value="Go" />
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
    <p class="searchtip" style="font-size: 90%">
    Enter search terms or a module, class or function name.
    </p>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             >index</a></li>
        <li class="right" >
          <a href="chapter02.html" title="DNA Variant Calling"
             >next</a> |</li>
        <li class="right" >
          <a href="index.html" title="Welcome to NGS 2014’s documentation!"
             >previous</a> |</li>
        <li><a href="index.html">ngs2014 Oct 2014 documentation</a> &raquo;</li> 
      </ul>
    </div>
    <div class="footer">
        &copy; Copyright 2014, T. Rausch.
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 1.2b1.
    </div>
  </body>
</html>