<!DOCTYPE html>
<html lang="en">
  <head>
    <title>SARS-CoV-2 variant calling</title>
  </head>
  <body>

    <h2>SARS-CoV-2 variant calling tutorial</h2>

    <h3>Tutorial URL: <a href="https://tobiasrausch.com/courses/bio.html">tobiasrausch.com/courses/bio.html</a></h3>

    <h3>Code repository: <a href="https://github.com/tobiasrausch/covid19">github.com/tobiasrausch/covid19</a></h3>
    
    <h3>Step 1: Alignment</h3>
    
    <p>In this tutorial we will analyze a positive SARS-CoV-2 sample that has been sequenced using an amplicon-based sequencing protocol.
      We will align the <a href="https://en.wikipedia.org/wiki/FASTQ_format">FASTQ</a> data to the <a href="https://www.ncbi.nlm.nih.gov/nuccore/1798174254">SARS-CoV-2 reference genome</a>, call variants and determine the viral lineage (i.e. Alpha, Beta, Gamma, Delta or Omicron). All the required tools and data sets have been pre-installed in a <a href="https://docs.conda.io/">conda</a> environment. Please open a terminal and load the conda environment.</p>

    <pre><code>
	export PATH=~/Desktop/covid19/conda/bin:${PATH}
	source activate covid19
    </code></pre>

    Next let's create a working directory for all the output files.
    <pre><code>
	mkdir ~/Desktop/covid19/tmp
	cd ~/Desktop/covid19/tmp
    </code></pre>

    The FASTQ files are in the data subdirectory and the SARS-CoV-2 reference is in the ref subdirectory.
    <pre><code>
	ls ../data/
	ls ../ref/NC_045512.2.fa
	head ../ref/NC_045512.2.fa
    </code></pre>

    We can use <a href="https://github.com/lh3/bwa">bwa</a> to align the sequencing data to the SARS-CoV-2 reference genome.
    <pre><code>
	bwa mem ../ref/NC_045512.2.fa ../data/Plate42B2/Plate42B2.R1.fastq.gz ../data/Plate42B2/Plate42B2.R2.fastq.gz > align.sam
	head align.sam
    </code></pre>

    The alignment is encoded in <a href="https://en.wikipedia.org/wiki/SAM_(file_format)">SAM format</a>. For mammalian-sized genomes, alignments are usually stored in a compressed binary format, called BAM, that can be indexed for random access so that, for instance, alignment viewers such as <a href="https://software.broadinstitute.org/software/igv/">IGV</a> can quickly visualize selected alignment regions. <a href="https://github.com/samtools/samtools">SAMtools</a> is a suite of tools that can be used to work with alignment files.

    <pre><code>
	samtools sort -o align.bam align.sam
	samtools index align.bam
    </code></pre>

    Once you have an alignment file it is always a good idea to look at the raw data in an alignment viewer such as <a href="https://software.broadinstitute.org/software/igv/">IGV</a>.
    <pre><code>
	../IGV_Linux_2.12.3/igv.sh -g ../ref/NC_045512.2.fa
    </code></pre>

    <h3>Step 2: Variant Calling</h3>

    Next, we want to transform the alignment into a list of variants. Hence, we are trying to identify positions, where the sequenced sample deviates from the SARS-CoV-2 reference genome. Variants are stored in <a href="https://en.wikipedia.org/wiki/Variant_Call_Format">VCF format</a>.
    <pre><code>
	bcftools mpileup -f ../ref/NC_045512.2.fa align.bam | bcftools call -mv -Ob -o calls.bcf
	bcftools view calls.bcf
	bcftools view calls.bcf | grep -A 2 -m 1 "^#CHROM"
    </code></pre>
    
    <h3>Step 3: Variant Annotation</h3>

    Tools such as <a href="https://www.ensembl.org/info/docs/tools/vep/index.html">the variant effect predictor (VEP)</a> can be used to determine the <a href="https://www.ensembl.org/info/genome/variation/prediction/predicted_data.html">consequences of genomic variants</a>. Please upload the below list of variants that you called from the alignment to the <a href="https://covid-19.ensembl.org/Tools/VEP">VEP online web application</a>.
    <pre><code>
	bcftools view calls.bcf | grep -v "^#" | cut -f 1-5
    </code></pre>

    Based on the VEP output, we will now try to manually identify the viral lineage (Alpha, Beta, Gamma, Delta, Omicron, etc.) using the predicted amino acid changes in the spike protein (S gene). A nice overview table of the different amino acid changes occurring in the different lineages is available on the <a href="https://outbreak.info/compare-lineages">outbreak.info</a> website.

    <h3>Step 4: Lineage classification</h3>
    After years of the SARS-CoV-2 pandemic, we have of course tools to automatically determine the viral lineage and these usually require a so-called consensus sequence of the alignment. Tools such as <a href="https://github.com/andersen-lab/ivar">iVar</a> can generate such a viral consensus genome.

    <pre><code>
	samtools mpileup -aa -A -d 0 -B -Q 0 align.bam | ivar consensus -t 0.9 -m 20 -n N -p cons
    </code></pre>


    This consensus sequence, we can use with <a href="https://github.com/cov-lineages/pangolin">pangolin</a> or <a href="https://clades.nextstrain.org/">Nextclade</a> to determine the lineage.

    <pre><code>
	source activate pangolin
	pangolin --outfile lineage.csv cons.fa
    </code></pre>
	
    <a href="https://clades.nextstrain.org/">Nextclade</a> has an online application that we can also use to upload the cons.fa file. Nextclade also integrates our sequence into a phylogenetic tree of publicly available SARS-CoV-2 sequences.
    
    
  </body>
</html>
