<!DOCTYPE html>
<html lang="en">
  <head>
    <BASE HREF="https://tobiasrausch.com/courses/cg/">
    <title>Analytical Methods in Cancer Genomics</title>
  </head>
  <body>
    
    <h2>Analytical Methods in Cancer Genomics</h2>

    <h3>Course Content</h3>

    This course will focus on the analysis of short-read and long-read sequencing data from cancer genomics studies. Bioinformatic concepts, tools and methods required to analyse tumor sequencing data will be introduced. Learning outcomes include an overview of the challenges in the study of cancer genomics, discovery and visualisation of copy-number and structural variants, understanding the principles of tumor purity, heterogeneity and ploidy and an overview of cancer epigenetics. The course covers different sequencing data modalities (short-reads vs. long-reads) and data types (bulk vs. single-cell). Practical data analysis sessions will complement the course.

    <h3>Schedule</h3>

    <ul>
      <li>Thursday 16th March, 2pm-3.30pm: Genome Variation, <a href="https://gear.embl.de/data/.slides/Lecture1_GenomeVariation.pdf">Slides</a></li>
      <li>Thursday 23th March, 2pm-3.30pm: Cancer Genomics</li>
      <li>Thursday 30th March, 2pm-3.30pm: Mutation Calling (Exercise Discussion)</li>
      
      <li>Thursday 6th April, 2pm-3.30pm: Cancer Epigenetics</li>
      <li>Thursday 13th April, 2pm-3.30pm: Structural Variants and copy-number variants</li>
      <li>Thursday 20th April, 2pm-3.30pm: Long reads in cancer genomics (Excercis Discussion)</li>
      <li>Thursday 27th April, 2pm-3.30pm: Genome assembly and pan-genome graphs</li>
      <li>Thursday 4th May, 2pm-3.30pm: Single-cell analysis</li>
      <li>Thursday 11th May, 2pm-3.30pm: Journal Club, Cancer Genomics Papers (Prague, Vinicna 7, room TBD).</li>
    </ul>

    <h3>Useful links</h3>
    
    Below are a couple of links to commonly used Bioinformatics tools in Cancer Genomics (certainly not comprehensive).
    <br>
    Next-generation sequencing analysis tutorials
    <ul>
      <li><a href="https://tobiasrausch.com/courses/vc/">NGS tutorial</a></li>
      <li><a href="https://github.com/ekg/alignment-and-variant-calling-tutorial">Alignment and variant calling</a></li>
    </ul>
    Commonly used alignment tools
    <ul>
      <li><a href="https://github.com/lh3/bwa">BWA</a></li>
      <li><a href="http://bowtie-bio.sourceforge.net/bowtie2/index.shtml">Bowtie2</a></li>
    </ul>
    Tools for working with alignment files (BAM files)
    <ul>
      <li><a href="https://github.com/samtools/htslib">HTSlib</a></li>
      <li><a href="https://github.com/samtools/samtools">SAMtools</a></li>
      <li><a href="https://github.com/arq5x/bedtools2">bedtools</a></li>
    </ul>
    Tools to compute read counts in windows
    <ul>
      <li><a href="https://github.com/brentp/mosdepth">mosdepth</a></li>
      <li><a href="https://github.com/tobiasrausch/alfred">alfred</a></li>
      <li><a href="https://github.com/samtools/samtools">SAMtools</a></li>
    </ul>
    Tools for short variant calling, i.e., point mutations (SNVs) and short insertions and deletions (InDels)
    <ul>
      <li><a href="https://github.com/Illumina/strelka">Strelka</a></li>
      <li><a href="https://github.com/freebayes/freebayes">FreeBayes</a></li>
    </ul>
    Tools for structural variant (SV) calling
    <ul>
      <li><a href="https://github.com/dellytools/delly">delly</a></li>
      <li><a href="https://github.com/arq5x/lumpy-sv">lumpy</a></li>
    </ul>
    Tools for working with variant call files (VCF/BCF)
    <ul>
      <li><a href="https://github.com/samtools/htslib">HTSlib</a></li>
      <li><a href="https://github.com/samtools/bcftools">BCFtools</a></li>
    </ul>
  </body>
</html>
