<!DOCTYPE html>
<html lang="en">
  <head>
    <BASE HREF="https://tobiasrausch.com/courses/cg/">
    <title>Analytical Methods in Cancer Genomics</title>
  </head>
  <body>
    
    <h2>Analytical Methods in Cancer Genomics</h2>

    <h3>Course Content</h3>

    This course will focus on the analysis of data from cancer genomics studies. Bioinformatic concepts, tools and methods required to analyse tumor sequencing data will be introduced. Learning outcomes include an overview of the challenges in the study of cancer genomics, discovery and visualisation of copy-number and structural variants, understanding the principles of tumor purity, heterogeneity and ploidy and an overview of cancer epigenetics. Practical data analysis sessions in R or python and using shell scripts will complement the course, including an overview of machine learning techniques.

    <h3>Schedule</h3>

    <ul>
      <li>Tuesday 6th Apr, 2pm-3pm: Introduction to Cancer Genomics, <a href="https://gear.embl.de/data/.slides/cancer_genomics_rausch.pdf">Slides</a></li>
      <li>Tuesday 13th Apr, 2pm-3pm: Genomic Rearrangements, <a href="https://gear.embl.de/data/.slides/structural_variants_rausch.pdf">Slides</a></li>
      <li>Tuesday 20th Apr, 2pm-3pm: Version control, electronic notebooks, markdown and package managers (Exercise discussion), <a href="https://gear.embl.de/data/.slides/exercise_rausch.pdf">Slides</a></li>
      <li>Tuesday 27th Apr, 2pm-3pm: Cancer Epigenetics, <a href="https://gear.embl.de/data/.slides/epigenetics_rausch.pdf">Slides</a></li>
      <li>Tuesday 4th May, 2pm-3pm: <a href="https://tobiasrausch.com/courses/cg/exploration.html">Data Exploration</a>, <a href="https://tobiasrausch.com/courses/cg/supervised.html">Supervised Learning</a>, <a href="https://tobiasrausch.com/courses/cg/regression.html">Regression using R Statistics</a> (<a href="https://gear.embl.de/data/.slides/machine_learning_rausch.pdf">Slides</a>)</li>
      <li>Tuesday 11th May, 2pm-3pm: <a href="https://docs.google.com/spreadsheets/d/19jn-hIXZe5H8rIk4h-bxis6fzSwDfn1iuuHxwQaG6-g/edit?usp=sharing">Presentations by students</a></li>
      <li>Tuesday 18th May, 2pm-3pm: Presentations by students</li>
      <li>Tuesday 25th May, 2pm-3pm: Presentations by students</li>
    </ul>

    <h3>Cancer Genomics Data Analysis Exercise (due date 4th May 2021, 2pm CET)</h3>
    In this exercise we want to analyze a cancer genomics sample (a paired tumor-control sample set).
    This is a simulated Medulloblastoma sample with a matched control.
    You can download the data set <a href="https://gear.embl.de/data/.exercise/">here</a>.
    The main objective of this exercise is to align the data to the reference genome, to sort and index the alignments and to generate a read-depth plot, as discussed in the lectures.
    Please note that I downsampled the dataset and I also just kept the data for chromosome X from 20Mbp to 40Mbp (GRCh37/hg19 coordinates) because otherwise all analysis take a lot of time.
    Once you have generated the alignment in BAM format you can subset the BAM to the region of interest using `samtools view -b input.bam chrX:20000000-40000000 > output.bam`.
    <br>
    In order to learn and get to know version control softwares (git) please create yourself a <a href="https://github.com/">Github account</a>, create a new repository and write-up your analysis pipeline using <a href="https://guides.github.com/features/mastering-markdown/">Markdown</a>.
    You can, of course, in additon use the Github repository to store your analysis scripts in your favorite language, i.e., this could be bash scripts, <a href="https://snakemake.readthedocs.io/en/stable/">Snakemake</a> or <a href="https://www.nextflow.io/">Nextflow</a> pipelines, <a href="https://www.r-project.org/">R</a> or <a href="https://www.python.org/">python</a> scripts.
    Likewise feel free to check-in a Makefile or a requirements file for <a href="https://conda.io/projects/conda/en/latest/user-guide/getting-started.html">Bioconda</a> if you use these to install tools.
    At the very minimum the repository should contain the produced read-depth plot and a README.md file that explains the steps you have executed to generate the read-depth plot.
    Once you are done please email me the repository link using the email address which you used to register for the course.
    <br>
    **Optional**: Once you have successfully computed a read-depth plot you may also want to call structural variants and overlay these with the read-depth plot or call germline variants and annotate these with functional consequences (<a href="https://gear.embl.de/data/.slides/exercise_rausch.pdf">Slides</a>).


    <h3>Machine Learning Programming Exercise (due date 8th June 2021)</h3>
    In 2015, the 1000 Genomes Project published a map of genomic structural variants for 2504 individuals sampled from different parts of the world (<a href="https://www.nature.com/articles/nature15394">Paper</a>). The simplest structural variant is a deletion, where a long stretch of DNA is missing in one sample with respect to the human reference genome. For the 1000 Genomes Project we were specifically interested in cataloging these deletions, basically finding out where they are in the genome and how many people among the 2504 individuals share the same deletion. As an example, you can see below a deletion visualized in <a href="https://software.broadinstitute.org/software/igv/">IGV</a> where one individual (shown in the upper panel) is a heterozygous carrier of the deletion (marked as a red bar at the top) and the lower panel shows an individual that is not a carrier of this deletion.
    <br>
    <img src="DEL00072515.png" alt="Deletion">
    The alignment signals that structural variant prediction algorithms evaluate are a change in read-depth because one haplotype is lost (see coverage drop in IGV), read-pairs that span the deletion and have an insert size larger than expected (read pairs colored in red in IGV) and reads that have a prefix alignment before the start of the deletion or a suffix alignment after the deletion (the colorful bases in IGV are the trimmed portions of reads not mapped because of the deletion).
    <br>
    The different research groups in the 1000 Genomes project predicted genomic deletions in these 2504 samples. As usual, some predictions (hopefully the majority) will be true and other predicted deletions will be false for various reasons (mis-mappings, bugs in the algorithms, incomplete reference sequence, etc.). I have annotated a subset of the deletion predictions where we have prior validation information on as likely true (status = 1) and as likely false (status = 0). I then computed a list of features for these deletions that you can later use in your machine learning models to predict for the unlabeled deletions whether these are likely true or likely false. Below is a brief explanation for each feature.
    <br>
    Features
    <br>
    <ul>
      <li>size: Size of the deletion</li>
      <li>vac: How many alleles have the deletion (2504 diploid samples -> 5008 alleles in total)</li>
      <li>vaf: Allele frequency (simply vac/5008)</li>
      <li>pass: 1 - Deletion prediction passed all algorithm filters, 0 – deletion did not pass all filters</li>
      <li>missingrate: Fraction of samples not genotyped for this deletion</li>
      <li>precise: 1 - The deletion has single-nucleotide resolution, 0 - only approximate breakpoints are known</li>
      <li>ci: Confidence interval size around the deletion breakpoints (how sure is the algorithm about the breakpoint)</li>
      <li>ce: DNA sequence entropy of the deletion</li>
      <li>refgq: Average genotype quality of homozygous reference samples</li>
      <li>altgq: Average genotype quality of deletion carrier samples</li>
      <li>rdratio: Read-depth ratio of deletion carriers compared to deletion non-carriers</li>
      <li>altratio: Average fraction of reads supporting the deletion in a carrier sample</li>
      <li>refratio: Average fraction of reads supporting the deletion in a non-carrier sample</li>
      <li>maxaltratio: Maximum fraction of reads supporting the deletion in a carrier sample</li>
      <li>status: 1 – likely true deletion, 0 – likely false deletion, NA – unlabeled (deletion could be true or false)</li>
    </ul>
    <br>
    My very simple example approach to predict the status of unlabeled deletions (true/false) is based on the genotyping quality of deletion carriers (altgq). An Rscript for this approach is available here. Please change this Rscript with your favorite machine learning technique and ideally improve my very simplistic decision tree approach. Once you are done please commit your Rscript to your GitHub repository from the Cancer Genomics exercise in a new subfolder. Please also include some comments either as a separate Markdown file or within the code. If you prefer to use a different programming language to run your machine learning models you are welcome to do so. The deadline for this exercise is 8th June 2021, please email me again your GitHub repository URL once you are done. Thanks.

    
    <h3>Useful links</h3>
    
    Below are a couple of links to commonly used Bioinformatics tools in Cancer Genomics (certainly not comprehensive).
    <br>
    Next-generation sequencing analysis tutorials
    <ul>
      <li><a href="https://tobiasrausch.com/courses/vc/">NGS tutorial</a></li>
      <li><a href="https://github.com/ekg/alignment-and-variant-calling-tutorial">Alignment and variant calling</a></li>
    </ul>
    Commonly used alignment tools
    <ul>
      <li><a href="https://github.com/lh3/bwa">BWA</a></li>
      <li><a href="http://bowtie-bio.sourceforge.net/bowtie2/index.shtml">Bowtie2</a></li>
    </ul>
    Tools for working with alignment files (BAM files)
    <ul>
      <li><a href="https://github.com/samtools/htslib">HTSlib</a></li>
      <li><a href="https://github.com/samtools/samtools">SAMtools</a></li>
      <li><a href="https://github.com/arq5x/bedtools2">bedtools</a></li>
    </ul>
    Tools to compute read counts in windows
    <ul>
      <li><a href="https://github.com/brentp/mosdepth">mosdepth</a></li>
      <li><a href="https://github.com/tobiasrausch/alfred">alfred</a></li>
      <li><a href="https://github.com/samtools/samtools">SAMtools</a></li>
    </ul>
    Tools for short variant calling (point mutations and short insertions and deletions)
    <ul>
      <li><a href="https://github.com/Illumina/strelka">Strelka</a></li>
      <li><a href="https://github.com/freebayes/freebayes">FreeBayes</a></li>
    </ul>
    Tools for structural variant calling
    <ul>
      <li><a href="https://github.com/dellytools/delly">delly</a></li>
      <li><a href="https://github.com/arq5x/lumpy-sv">lumpy</a></li>
    </ul>
    Tools for working with variant call files (VCF/BCF)
    <ul>
      <li><a href="https://github.com/samtools/htslib">HTSlib</a></li>
      <li><a href="https://github.com/samtools/bcftools">BCFtools</a></li>
    </ul>
  </body>
</html>
