<!DOCTYPE html>
<html lang="en">
  <head>
    <BASE HREF="https://tobiasrausch.com/courses/cg/">
    <title>Analytical Methods in Cancer and Population Genomics</title>
  </head>
  <body>
    
    <h2>Analytical Methods in Cancer and Population Genomics</h2>

    <h3>Course Content</h3>

    This part of the course will focus on the analysis of short-read and long-read sequencing data from population and cancer genomics studies. Bioinformatic concepts, tools and methods required to analyse (tumor) sequencing data will be introduced. Learning outcomes include an overview of the challenges in the study of (cancer) genomic data sets, discovery and visualisation of copy-number and structural variants, understanding the principles of tumor purity, heterogeneity and ploidy and an overview of cancer epigenetics. The course covers different sequencing data modalities (short-reads vs. long-reads) and data types (bulk vs. single-cell). Practical data analysis sessions will complement the course.

    <h3>Schedule</h3>

    <ul>
      <li>Friday 22th November, 12:20pm-2pm: Lecture1 - Introduction to Cancer Genomics (Zoom), <a href="https://gear-genomics.embl.de/data/.slides/Lecture1_CancerGenomics.pdf">Slides</a></li>
    </ul>

    <h3>Useful links</h3>
    
    Below are a couple of links to commonly used Bioinformatics tools in population and cancer genomics (certainly not comprehensive).
    <br>
    Next-generation sequencing analysis tutorials
    <ul>
      <li><a href="https://github.com/ekg/alignment-and-variant-calling-tutorial">Alignment and variant calling</a></li>
      <li><a href="https://github.com/tobiasrausch/vc">Short-read structural variant calling tutorial</a></li>
      <li><a href="https://github.com/tobiasrausch/sv">Long-read structural variant calling tutorial</a></li>
    </ul>
    Commonly used alignment tools
    <ul>
      <li><a href="https://github.com/lh3/bwa">BWA</a></li>
      <li><a href="https://github.com/lh3/minimap2">Minimap2</a></li>
      <li><a href="http://bowtie-bio.sourceforge.net/bowtie2/index.shtml">Bowtie2</a></li>
    </ul>
    Tools for working with alignment files (BAM files)
    <ul>
      <li><a href="https://github.com/samtools/htslib">HTSlib</a></li>
      <li><a href="https://github.com/samtools/samtools">SAMtools</a></li>
      <li><a href="https://github.com/arq5x/bedtools2">bedtools</a></li>
    </ul>
    Tools for working with variant files (VCF/BCF files)
    <ul>
      <li><a href="https://github.com/samtools/bcftools">BCFtools</a></li>
      <li><a href="https://brentp.github.io/cyvcf2/">cyvcf2 python library</a></li>
      <li><a href="https://github.com/samtools/htslib">HTSlib</a></li>
    </ul>
    Tools to compute read counts in windows
    <ul>
      <li><a href="https://github.com/brentp/mosdepth">mosdepth</a></li>
      <li><a href="https://github.com/tobiasrausch/alfred">alfred</a></li>
      <li><a href="https://github.com/samtools/samtools">SAMtools</a></li>
      <li><a href="https://github.com/dellytools/delly">delly</a></li>
    </ul>
    Tools for variant calling, i.e., point mutations (SNVs) and short insertions and deletions (InDels)
    <ul>
      <li><a href="https://github.com/Illumina/strelka">Strelka</a></li>
      <li><a href="https://github.com/freebayes/freebayes">FreeBayes</a></li>
    </ul>
    Tools for structural variant (SV) calling
    <ul>
      <li><a href="https://github.com/dellytools/delly">delly</a></li>
      <li><a href="https://github.com/arq5x/lumpy-sv">lumpy</a></li>
    </ul>
    Working with count matrices
    <ul>
      <li><a href="http://bioconductor.org/packages/devel/bioc/vignettes/DESeq2/inst/doc/DESeq2.html">DESeq2 Tutorial</a></li>
      <li><a href="https://www.gsea-msigdb.org/gsea/">Gene set enrichment analysis (GSEA)</a></li>
      <li><a href="https://maayanlab.cloud/Enrichr/">Enrichr</a></li>
    </ul>
    <ul>
  </body>
</html>
