<!DOCTYPE html>
<html lang="en">
  <head>
    <BASE HREF="https://tobiasrausch.com/courses/cg/">
    <title>Analytical Methods in Cancer Genomics</title>
  </head>
  <body>
    
    <h2>Analytical Methods in Cancer Genomics</h2>

    <h3>Course Content</h3>

    This course will focus on the analysis of data from cancer genomics studies. Bioinformatic concepts, tools and methods required to analyse tumor sequencing data will be introduced. Learning outcomes include an overview of the challenges in the study of cancer genomics, discovery and visualisation of copy-number and structural variants, understanding the principles of tumor purity, heterogeneity and ploidy and an overview of cancer epigenetics. Practical data analysis sessions will complement the course.

    <h3>Schedule</h3>

    <ul>
      <li>Tuesday 8th March, 1pm-2.30pm: Genome Variation, <a href="https://gear.embl.de/data/.slides/Lecture1_GenomeVariation.pdf">Slides</a></li>
      <li>Tuesday 15th March, 1pm-2.30pm: Cancer Genomics, <a href="https://gear.embl.de/data/.slides/Lecture2_CancerGenomics.pdf">Slides</a></li>
      <li>Tuesday 22th March, 1pm-2.30pm: Variant Calling Exercise, <a href="https://gear.embl.de/data/.slides/Lecture3_MultiOmics.pdf">Slides</a></li>
      <li>Tuesday 29th March, 1pm-2.30pm: Cancer Epigenetics, <a href="https://gear.embl.de/data/.slides/Lecture4_Epigenetics.pdf">Slides</a></li>
      <li>Tuesday 5th April, 1pm-2.30pm: Structural Variants, <a href="https://gear.embl.de/data/.slides/Lecture5_StructuralVariants.pdf">Slides</a></li>
      <li>Tuesday 12th April, 1pm-2.30pm: Machine Learning, <a href="https://gear.embl.de/data/.slides/Lecture6_MachineLearning.pdf">Slides</a></li>
    </ul>

    <h3>Exercise 1: Sars-CoV-2 Lineage Determination (due date 5th April 2022)</h3>

    Align the FASTQ files to the SARS-CoV-2 reference genome, call variants and annotate variants. Determine the most likely SARS-CoV-2 lineage (Alpha, Beta, Gamma, Delta or Omicron).

    <ul>
      <li>Data set 1, <a href="https://gear.embl.de/data/.slides/Plate42B2.R1.fastq.gz">Read1</a></li>
      <li>Data set 1, <a href="https://gear.embl.de/data/.slides/Plate42B2.R2.fastq.gz">Read2</a></li>
      <li>Data set 2, <a href="https://gear.embl.de/data/.slides/Plate135H10.R1.fastq.gz">Read1</a></li>
      <li>Data set 2, <a href="https://gear.embl.de/data/.slides/Plate135H10.R2.fastq.gz">Read2</a></li>
    </ul>
    
    <h3>Exercise 2: Cancer Genomics Data Analysis (due date 19th April 2022)</h3>
    In this exercise we want to analyze a cancer genomics sample (a paired tumor-control sample set). This is a simulated Medulloblastoma sample with a matched control.
    You can download the data set <a href="https://gear.embl.de/data/.exercise/">here</a>.
    The main objective of this exercise is to align the data to the human reference genome (<a href="https://hgdownload.soe.ucsc.edu/goldenPath/hg19/bigZips/hg19.fa.gz">https://hgdownload.soe.ucsc.edu/goldenPath/hg19/bigZips/hg19.fa.gz</a>), to sort and index the alignments and to generate a read-depth plot, as discussed in the lectures.
    Please note that I downsampled the dataset and I also just kept the data for chromosome X from 20Mbp to 40Mbp (GRCh37/hg19 coordinates) because otherwise all analysis take a lot of time for a human genome.
    Once you have generated the alignment in BAM format you can subset the BAM to the region of interest using `samtools view -b input.bam chrX:20000000-40000000 > output.bam`.
    <br>
    Please write-up your analysis pipeline using <a href="https://guides.github.com/features/mastering-markdown/">Markdown</a>.
    You can, of course, in additon use the Github repository to store your analysis scripts in your favorite language, i.e., this could be bash scripts, <a href="https://snakemake.readthedocs.io/en/stable/">Snakemake</a> or <a href="https://www.nextflow.io/">Nextflow</a> pipelines, <a href="https://www.r-project.org/">R</a> or <a href="https://www.python.org/">python</a> scripts.
    Likewise feel free to check-in a Makefile or a requirements file for <a href="https://conda.io/projects/conda/en/latest/user-guide/getting-started.html">Bioconda</a> if you use these to install tools.
    At the very minimum the repository should contain the produced read-depth plot and a README.md file that explains the steps you have executed to generate the read-depth plot.
    Once you are done please email me again the repository link. Thanks.
    <br>
    **Optional**: Once you have successfully computed a read-depth plot you may also want to call structural variants and overlay these with the read-depth plot or call germline variants and annotate these with functional consequences.
        
    <h3>Useful links</h3>
    
    Below are a couple of links to commonly used Bioinformatics tools in Cancer Genomics (certainly not comprehensive).
    <br>
    Next-generation sequencing analysis tutorials
    <ul>
      <li><a href="https://tobiasrausch.com/courses/vc/">NGS tutorial</a></li>
      <li><a href="https://github.com/ekg/alignment-and-variant-calling-tutorial">Alignment and variant calling</a></li>
    </ul>
    Commonly used alignment tools
    <ul>
      <li><a href="https://github.com/lh3/bwa">BWA</a></li>
      <li><a href="http://bowtie-bio.sourceforge.net/bowtie2/index.shtml">Bowtie2</a></li>
    </ul>
    Tools for working with alignment files (BAM files)
    <ul>
      <li><a href="https://github.com/samtools/htslib">HTSlib</a></li>
      <li><a href="https://github.com/samtools/samtools">SAMtools</a></li>
      <li><a href="https://github.com/arq5x/bedtools2">bedtools</a></li>
    </ul>
    Tools to compute read counts in windows
    <ul>
      <li><a href="https://github.com/brentp/mosdepth">mosdepth</a></li>
      <li><a href="https://github.com/tobiasrausch/alfred">alfred</a></li>
      <li><a href="https://github.com/samtools/samtools">SAMtools</a></li>
    </ul>
    Tools for short variant calling (point mutations and short insertions and deletions)
    <ul>
      <li><a href="https://github.com/Illumina/strelka">Strelka</a></li>
      <li><a href="https://github.com/freebayes/freebayes">FreeBayes</a></li>
    </ul>
    Tools for structural variant calling
    <ul>
      <li><a href="https://github.com/dellytools/delly">delly</a></li>
      <li><a href="https://github.com/arq5x/lumpy-sv">lumpy</a></li>
    </ul>
    Tools for working with variant call files (VCF/BCF)
    <ul>
      <li><a href="https://github.com/samtools/htslib">HTSlib</a></li>
      <li><a href="https://github.com/samtools/bcftools">BCFtools</a></li>
    </ul>
  </body>
</html>
