<!DOCTYPE html>
<html lang="en">
  <head>
    <BASE HREF="https://tobiasrausch.com/courses/cg/">
    <title>Analytical Methods in Cancer and Population Genomics</title>
  </head>
  <body>
    
    <h2>Analytical Methods in Cancer and Population Genomics</h2>

    <h3>Course Content</h3>

    This part of the course will focus on the analysis of short-read and long-read sequencing data from population and cancer genomics studies. Bioinformatic concepts, tools and methods required to analyse (tumor) sequencing data will be introduced. Learning outcomes include an overview of the challenges in the study of (cancer) genomic data sets, discovery and visualisation of copy-number and structural variants, understanding the principles of tumor purity, heterogeneity and ploidy and an overview of cancer epigenetics. The course covers different sequencing data modalities (short-reads and long-reads) and data types (bulk and single-cell). Practical data analysis sessions will complement the course.

    <h3>Exercise 1: Variant Calling (due date 9th December 2025)</h3>

    The goal of this exercise is to create a simple variant calling workflow for human genomic sequencing data. The workflow should contain steps to align the FASTQ files to the human reference genome (<a href="https://github.com/lh3/bwa">bwa</a>), call variants (<a href="https://samtools.github.io/bcftools/howtos/variant-calling.html">bcftools</a>) and annotate all single-nucleotide variants with (<a href="https://www.ensembl.org/Tools/VEP">VEP</a>). For this excercise you can ignore all insertions and deletions, only single-nucleotide variants (SNVs) should be considered. The data has been subsampled to chr7 to speed-up all analysis. Once you have finished the exercise, please submit your answer using this <a href="https://docs.google.com/forms/d/e/1FAIpQLScWe3EjOV3ELqrcI2-BsgvypxssgsbUvVPxWPKYVl66touowg/viewform?usp=sharing&ouid=107866491445506675294">Google Form</a>.
    
    <ul>
      <li>Chromosome 7 human reference, <a href="https://hgdownload.soe.ucsc.edu/goldenPath/hg38/chromosomes/chr7.fa.gz">chr7</a></li>
      <li>FASTQ of read1, <a href="https://gear-genomics.embl.de/data/.slides/R1.fastq.gz">Read1</a></li>
      <li>FASTQ of read2, <a href="https://gear-genomics.embl.de/data/.slides/R2.fastq.gz">Read2</a></li>
    </ul>


    <h3>Schedule</h3>

    <ul>
      <li>Tuesday 2nd December: Lecture 1 - Introduction to Cancer Genomics (Zoom), <a href="https://gear-genomics.embl.de/data/.slides/CancerGenomics.pdf">Slides</a></li>
    </ul>

    <h3>Useful links</h3>
    
    Below are a couple of links to commonly used Bioinformatics tools in population and cancer genomics (certainly not comprehensive).
    <br>
    Next-generation sequencing analysis tutorials
    <ul>
      <li><a href="https://github.com/ekg/alignment-and-variant-calling-tutorial">Alignment and variant calling</a></li>
      <li><a href="https://github.com/tobiasrausch/vc">Short-read structural variant calling tutorial</a></li>
      <li><a href="https://github.com/tobiasrausch/sv">Long-read structural variant calling tutorial</a></li>
    </ul>
    Commonly used alignment tools
    <ul>
      <li><a href="https://github.com/lh3/bwa">BWA</a></li>
      <li><a href="https://github.com/lh3/minimap2">Minimap2</a></li>
      <li><a href="http://bowtie-bio.sourceforge.net/bowtie2/index.shtml">Bowtie2</a></li>
    </ul>
    Tools for working with alignment files (BAM files)
    <ul>
      <li><a href="https://github.com/samtools/htslib">HTSlib</a></li>
      <li><a href="https://github.com/samtools/samtools">SAMtools</a></li>
      <li><a href="https://github.com/arq5x/bedtools2">bedtools</a></li>
    </ul>
    Tools for working with variant files (VCF/BCF files)
    <ul>
      <li><a href="https://github.com/samtools/bcftools">BCFtools</a></li>
      <li><a href="https://brentp.github.io/cyvcf2/">cyvcf2 python library</a></li>
      <li><a href="https://github.com/samtools/htslib">HTSlib</a></li>
    </ul>
    Tools to compute read counts in windows
    <ul>
      <li><a href="https://github.com/brentp/mosdepth">mosdepth</a></li>
      <li><a href="https://github.com/tobiasrausch/alfred">alfred</a></li>
      <li><a href="https://github.com/samtools/samtools">SAMtools</a></li>
      <li><a href="https://github.com/dellytools/delly">delly</a></li>
    </ul>
    Tools for variant calling, i.e., point mutations (SNVs) and short insertions and deletions (InDels)
    <ul>
      <li><a href="https://github.com/Illumina/strelka">Strelka</a></li>
      <li><a href="https://github.com/freebayes/freebayes">FreeBayes</a></li>
    </ul>
    Tools for structural variant (SV) calling
    <ul>
      <li><a href="https://github.com/dellytools/delly">delly</a></li>
      <li><a href="https://github.com/arq5x/lumpy-sv">lumpy</a></li>
      <li><a href="https://github.com/fritzsedlazeck/Sniffles">sniffles</a></li>
    </ul>
    Differential expression and gene set enrichment
    <ul>
      <li><a href="http://bioconductor.org/packages/devel/bioc/vignettes/DESeq2/inst/doc/DESeq2.html">DESeq2 Tutorial</a></li>
      <li><a href="https://www.gsea-msigdb.org/gsea/">Gene set enrichment analysis (GSEA)</a></li>
      <li><a href="https://maayanlab.cloud/Enrichr/">Enrichr</a></li>
    </ul>
    <ul>
  </body>
</html>
