<!DOCTYPE html>
<html>
  <body>
    
    <h2>Analytical Methods in Cancer Genomics</h2>

    <h3>Course Content</h3>

    This course will focus on the analysis of data from cancer genomics studies. Bioinformatic concepts, tools and methods required to analyse tumor sequencing data will be introduced. Learning outcomes include an overview of the challenges in the study of cancer genomics, discovery and visualisation of copy-number and structural variants, understanding the principles of tumor purity, heterogeneity and ploidy and an overview of cancer epigenetics. Practical data analysis sessions in R or python and using shell scripts will complement the course, including an overview of machine learning techniques.

    <h3>Schedule</h3>

    <ul>
      <li>Tuesday 6th Apr, 2pm-3pm: Introduction to Cancer Genomics, <a href="https://gear.embl.de/data/.slides/cancer_genomics_rausch.pdf">Slides</a></li>
      <li>Tuesday 13th Apr, 2pm-3pm: Genomic Rearrangements, <a href="https://gear.embl.de/data/.slides/structural_variants_rausch.pdf">Slides</a></li>
      <li>Tuesday 20th Apr, 2pm-3pm: Cancer Epigenetics, <a href="https://gear.embl.de/data/.slides/epigenetics_rausch.pdf">Slides</a></li>
    </ul>

    <h3>Cancer Genomics Data Analysis Exercise (due date 4th May 2021)</h3>
    In this exercise we want to analyze a cancer genomics sample (a paired tumor-control sample set). This is a simulated Medulloblastoma sample with a matched control. The main objective of this exercise is to align the data to the reference genome, to sort and index the alignments and to generate a read-depth plot, as discussed in the lectures. Please note that I downsampled the dataset and also just kept the data for chromosome X from 20Mbp to 40Mbp (GRCh37/hg19 coordinates) because otherwise all analysis take a lot of time. Once you have generated the alignment in BAM format you can subset the BAM to the region of interest using `samtools view -b input.bam chrX:20000000-40000000 > output.bam`.   In order to learn and get to know version control softwares (git) please create yourself a <a href="https://github.com/">Github account</a>, create a new repository and write-up your analysis pipeline using markdown. You can, of course, in additon use the Github repository to store your analysis scripts in your favorite language, i.e., this could be bash scripts, snakemake pipelines, R or python scripts or a requirements file for bioconda. At the very minimum the repository should contain the README.md file that explains the steps you have executed to generate the read-depth plot. Once you are done please email me the repository link using the email address which you you used to register for the course. **Optional**: Once you have successfully computed a read-depth plot you may also want to call structural variants and overlay these with the read-depth plot or call germline variants and annotate these with functional consequences (<a href="https://gear.embl.de/data/.slides/exercise_rausch.pdf">Slides</a></li>).
  </body>
</html>
