<!DOCTYPE html>
<html lang="en">
  <head>
    <BASE HREF="https://tobiasrausch.com/courses/cg/">
    <title>Analytical Methods in Cancer and Population Genomics</title>
  </head>
  <body>
    
    <h2>Analytical Methods in Cancer and Population Genomics</h2>

    <h3>Course Content</h3>

    This part of the course will focus on the analysis of short-read and long-read sequencing data from population and cancer genomics studies. Bioinformatic concepts, tools and methods required to analyse (tumor) sequencing data will be introduced. Learning outcomes include an overview of the challenges in the study of (cancer) genomic data sets, discovery and visualisation of copy-number and structural variants, understanding the principles of tumor purity, heterogeneity and ploidy and an overview of cancer epigenetics. The course covers different sequencing data modalities (short-reads vs. long-reads) and data types (bulk vs. single-cell). Practical data analysis sessions will complement the course.

    <h3>Schedule</h3>

    <ul>
      <li>Friday 22th November, 12:20pm-2pm: Lecture1 - Introduction to Cancer Genomics (Zoom, Part1), <a href="https://gear-genomics.embl.de/data/.slides/Lecture1_CancerGenomics.pdf">Slides</a></li>
      <li>Friday 29th November, 12:20pm-2pm: Lecture2 - Introduction to Cancer Genomics (Zoom, Part2), <a href="https://gear-genomics.embl.de/data/.slides/Lecture1_CancerGenomics.pdf">Slides</a></li>
      <li>Friday 6th December, 12:20pm-2pm: Lecture3 - Structural Variants, <a href="https://gear-genomics.embl.de/data/.slides/Lecture3_StructuralVariants.pdf">Slides</a></li>
      <li>Friday 13th December, 12:20pm-2pm: Lecture4 - Long-read and pan-genome graphs, <a href="https://gear-genomics.embl.de/data/.slides/Lecture4_LongReads.pdf">Slides</a></li>
      <li>Friday 20th December: No lecture!</li>
      <li>Friday 27th December: No lecture!</li>
      <li>Friday 3rd January: No lecture!</li>
      <li>Friday 10th January, 12:20pm-2pm: Lecture5 - Cancer Transcriptome, <a href="https://gear-genomics.embl.de/data/.slides/Lecture5_Transcriptome.pdf">Slides</a></li>
    </ul>

    <h3>Exercise 1: Variant Calling (due 10th January 2025)</h3>

    The goal of this exercise is to create a simple variant calling workflow for human sequencing data of a simulated rare disease case. The workflow should contain steps to align the FASTQ files to the human reference genome (<a href="https://github.com/lh3/bwa">bwa</a>), call variants (<a href="https://samtools.github.io/bcftools/howtos/variant-calling.html">bcftools</a>) and annotate all single-nucleotide variants with (<a href="https://www.ensembl.org/Tools/VEP">VEP</a>). For this excercise you can ignore all short insertions and deletions and also structural variants. We are only interested in single-nucleotide variants (SNVs).Once you have finished the exercise, please send me the identified pathogenic variant using this <a href="https://forms.gle/B3azUnBxLJ4tFXN6A">google form</a> and don't forget to provide your email address that you used to register for the course.

    <ul>
      <li>Chromosome 7 human reference, <a href="https://hgdownload.soe.ucsc.edu/goldenPath/hg38/chromosomes/chr7.fa.gz">chr7</a></li>
      <li>FASTQ of read1, <a href="https://gear-genomics.embl.de/data/.slides/R1.fastq.gz">Read1</a></li>
      <li>FASTQ of read2, <a href="https://gear-genomics.embl.de/data/.slides/R2.fastq.gz">Read2</a></li>
    </ul>

    <h3>Exercise 2: Cancer Genomics Data Analysis (due date 10th January 2025)</h3>
    In this exercise we want to analyze a cancer genomics sample, namely a paired tumor-normal sample pair.
    You can download the data set <a href="https://gear-genomics.embl.de/data/.exercise/">here</a>. 
    The main objective of this exercise is to align the data to the human reference genome using <a href="https://github.com/lh3/bwa">bwa</a>, to sort and index the alignments using <a href="https://github.com/samtools/samtools">SAMtools</a>, to generate a read-depth plot using <a href="https://github.com/tobiasrausch/alfred">alfred</a> and to call structural variants using <a href="https://github.com/dellytools/delly">delly</a>. Please note that I downsampled the dataset and I also just kept the data for chromosome X from 20Mbp to 40Mbp so please use the chrX FASTA file for alignment available <a href="https://gear-genomics.embl.de/data/.exercise/">here</a>. Once you have finished the exercise, please submit your answer using this <a href="https://forms.gle/ZuMQ96GxZ6SP22QFA">google form</a> and don't forget to provide your email address that you used to register for the course.
    <br>
    In short, the steps you need to do are:
    <ul>
      <li>Download the reference and FASTQ files using 'wget https://gear-genomics.embl.de/data/.exercise/chrX.fa' and so on.</li>
      <li>Index the reference file using 'bwa index chrX.fa'</li>
      <li>Align the FASTQ files to the reference using 'bwa mem chrX.fa wt.r1.fq.gz wt.r2.fq.gz' for the control (wt) and tumor (tu).</li>
      <li>Sort and index the alignments using samtools</li>
      <li>Count reads in 10,000bp windows using 'alfred count_dna -o wt.cov.gz wt.bam' for the control and the tumor.</li>
      <li>Merge the count tables row-by-row using the shell or your favorite programming language.</li>
      <li>Call structural variants using 'delly call -g chrX.fa -o delly.bcf tu.bam wt.bam'.</li>
    </ul>

    
    <h3>Useful links</h3>
    
    Below are a couple of links to commonly used Bioinformatics tools in population and cancer genomics (certainly not comprehensive).
    <br>
    Next-generation sequencing analysis tutorials
    <ul>
      <li><a href="https://github.com/ekg/alignment-and-variant-calling-tutorial">Alignment and variant calling</a></li>
      <li><a href="https://github.com/tobiasrausch/vc">Short-read structural variant calling tutorial</a></li>
      <li><a href="https://github.com/tobiasrausch/sv">Long-read structural variant calling tutorial</a></li>
    </ul>
    Commonly used alignment tools
    <ul>
      <li><a href="https://github.com/lh3/bwa">BWA</a></li>
      <li><a href="https://github.com/lh3/minimap2">Minimap2</a></li>
      <li><a href="http://bowtie-bio.sourceforge.net/bowtie2/index.shtml">Bowtie2</a></li>
    </ul>
    Tools for working with alignment files (BAM files)
    <ul>
      <li><a href="https://github.com/samtools/htslib">HTSlib</a></li>
      <li><a href="https://github.com/samtools/samtools">SAMtools</a></li>
      <li><a href="https://github.com/arq5x/bedtools2">bedtools</a></li>
    </ul>
    Tools for working with variant files (VCF/BCF files)
    <ul>
      <li><a href="https://github.com/samtools/bcftools">BCFtools</a></li>
      <li><a href="https://brentp.github.io/cyvcf2/">cyvcf2 python library</a></li>
      <li><a href="https://github.com/samtools/htslib">HTSlib</a></li>
    </ul>
    Tools to compute read counts in windows
    <ul>
      <li><a href="https://github.com/brentp/mosdepth">mosdepth</a></li>
      <li><a href="https://github.com/tobiasrausch/alfred">alfred</a></li>
      <li><a href="https://github.com/samtools/samtools">SAMtools</a></li>
      <li><a href="https://github.com/dellytools/delly">delly</a></li>
    </ul>
    Tools for variant calling, i.e., point mutations (SNVs) and short insertions and deletions (InDels)
    <ul>
      <li><a href="https://github.com/Illumina/strelka">Strelka</a></li>
      <li><a href="https://github.com/freebayes/freebayes">FreeBayes</a></li>
    </ul>
    Tools for structural variant (SV) calling
    <ul>
      <li><a href="https://github.com/dellytools/delly">delly</a></li>
      <li><a href="https://github.com/arq5x/lumpy-sv">lumpy</a></li>
    </ul>
    Working with count matrices
    <ul>
      <li><a href="http://bioconductor.org/packages/devel/bioc/vignettes/DESeq2/inst/doc/DESeq2.html">DESeq2 Tutorial</a></li>
      <li><a href="https://www.gsea-msigdb.org/gsea/">Gene set enrichment analysis (GSEA)</a></li>
      <li><a href="https://maayanlab.cloud/Enrichr/">Enrichr</a></li>
    </ul>
    <ul>
  </body>
</html>
