<!DOCTYPE html>
<html lang="en">
  <head>
    <title>SARS-CoV-2 variant calling</title>
  </head>
  <body>

    <h2>SARS-CoV-2 variant calling tutorial</h2>

    <h3>Tutorial URL: <a href="https://tobiasrausch.com/courses/bio.html">tobiasrausch.com/courses/bio.html</a></h3>

    <h3>Step 1: Alignment</h3>
    
    <p>In this tutorial we will analyze a positive SARS-CoV-2 sample that has been sequenced using an amplicon-based sequencing protocol.
      We will align the <a href="https://en.wikipedia.org/wiki/FASTQ_format">FASTQ</a> data to the <a href="https://www.ncbi.nlm.nih.gov/nuccore/1798174254">SARS-CoV-2 reference genome</a>, call variants and determine the viral lineage (i.e. Alpha, Beta, Gamma, Delta or Omicron). All the required tools and data sets have been pre-installed in a <a href="https://docs.conda.io/">conda</a> environment. Please open a terminal and load the conda environment.</p>

    <pre><code>
	export PATH=/home/ubuntu/Desktop/covid19/conda/bin:${PATH}
	source activate covid19
    </code></pre>

    Next let's create a working directory for all the output files.
    <pre><code>
	mkdir ~/Desktop/covid19/tmp
	cd ~/Desktop/covid19/tmp
    </code></pre>
	
  </body>
</html>
